{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook best viewed here: https://nbviewer.jupyter.org"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebooke encompasses\n",
    "- sending files to start varscan_pipeline on server\n",
    "- refiltering varscan output to get variety-specific SNPs\n",
    "- recalculating RD\n",
    "- maf filtering\n",
    "- LD pruning to get SNPs for structure estimation in GEA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pythonimports import *"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# copy over fastq and md5 files to compute canada server"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "352"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DIR = '/data/fastq/mengmeng/CoAdapTree_DouglasFir/received_2019_Sep10'\n",
    "fastqs = fs(DIR, pattern='.fastq')\n",
    "len(fastqs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "cmdtext = op.join(DIR, 'cp_to_graham_cmds.txt')\n",
    "with open(cmdtext, 'w') as o:\n",
    "    cmds = []\n",
    "    for fastq in fastqs:\n",
    "        cmds.append(f'rsync -avz {fastq} graham:/scratch/lindb/DF_pooled/')\n",
    "    o.write(\"%s\" % '\\n'.join(cmds))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "56 56\n"
     ]
    }
   ],
   "source": [
    "lview, dview = get_client('default')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "def exe(cmd):\n",
    "    import os\n",
    "    os.system(cmd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "352"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(cmds)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "61\n",
      "352\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-11-096e3f4e5dca>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mjobs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmake_jobs\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcmds\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mexe\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlview\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mwatch_async\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mjobs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/data/programs/brandon_anaconda3/mypy/pythonimports/pythonimports.py\u001b[0m in \u001b[0;36mwatch_async\u001b[0;34m(jobs, phase)\u001b[0m\n\u001b[1;32m    186\u001b[0m     \u001b[0mcount\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    187\u001b[0m     \u001b[0;32mwhile\u001b[0m \u001b[0mcount\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mjobs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 188\u001b[0;31m         \u001b[0mtime\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msleep\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m5\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    189\u001b[0m         \u001b[0mcount\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    190\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mj\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mjobs\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "jobs = make_jobs(cmds, exe, lview)\n",
    "watch_async(jobs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/data/fastq/mengmeng/CoAdapTree_DouglasFir/received_2019_Sep10/cp_to_graham_cmds.txt'"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cmdtext"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "for j in jobs:\n",
    "    x = j.r"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "needed = []\n",
    "for cmd in cmds:\n",
    "    fq = op.basename(cmd.split()[2])\n",
    "    for x in ['NS.1195.001.D707---D504.DF_p54_cap25_kit3_R1.fastq.gz',\n",
    "              'NS.1195.001.D707---D504.DF_p54_cap25_kit3_R2.fastq.gz',\n",
    "              'NS.1195.001.D707---D505.DF_p85_cap27_kit3_R1.fastq.gz']:\n",
    "        if fq == x:\n",
    "            needed.append(cmd)\n",
    "len(needed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3\n",
      "3\n"
     ]
    }
   ],
   "source": [
    "jobs = make_jobs(needed, exe, lview)\n",
    "watch_async(jobs)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# filter output based on variety\n",
    "\n",
    "I ran all pops through pipeline at the same time, but we want to also run analyses on each variety (coastal and interior) so we need to refilter snps."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "56 56\n"
     ]
    }
   ],
   "source": [
    "lview,dview = get_client()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "metadata": {},
   "outputs": [],
   "source": [
    "# modified from filter_VariantsToTable.py to only pull out baseline-filtered snps based on variety\n",
    "# modifications are marked with ########## (other than imports)\n",
    "def pklload(path):\n",
    "    import pickle\n",
    "    pkl = pickle.load(open(path, 'rb'))\n",
    "    return pkl\n",
    "dview['pklload'] = pklload\n",
    "\n",
    "def get_varscan_names(df, tablefile):                                          ############ added tablefile arg\n",
    "    \"\"\"Convert generic sample/pool names from varscan to something meaningful.\"\"\"\n",
    "    print('renaming varscan columns ...')\n",
    "    import os \n",
    "    pool = os.path.basename(os.path.dirname(os.path.dirname(os.path.dirname(tablefile))))          ############ added\n",
    "    parentdir = os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(tablefile))))      ############\n",
    "    \n",
    "    # get order of samps used to create varscan cmds (same order as datatable)\n",
    "    samps = pklload(os.path.join(parentdir, f'{pool}/pkl_files/poolsamps.pkl'))[pool]                 ############ \n",
    "    # create a list of names that varscan gives by default\n",
    "    generic = ['Sample%s' % (i+1) for i in range(len(samps))]\n",
    "    # create a map between generic and true samp names\n",
    "    dic = dict((gen, samp) for (gen, samp) in zip(generic, samps))\n",
    "    # rename the columns in df\n",
    "    cols = []\n",
    "    for col in df:\n",
    "        if '.' in col:\n",
    "            gen, rest = col.split(\".\")\n",
    "            samp = dic[gen]\n",
    "            col = '.'.join([samp, rest])\n",
    "        cols.append(col)\n",
    "    df.columns = cols\n",
    "    return df\n",
    "dview['get_varscan_names'] = get_varscan_names\n",
    "\n",
    "def load_data(tablefile, variety):\n",
    "    \"\"\"\n",
    "    Load the VariantsToTable output.\n",
    "    \n",
    "    Positional arguments:\n",
    "    tablefile - path to VariantsToTable output - used to find ploidy etc\n",
    "    \n",
    "    Returns:\n",
    "    df - pandas.dataframe; VariantsToTable output\n",
    "    tf - basename of tablefile\n",
    "    \"\"\"\n",
    "    import os\n",
    "    import pandas \n",
    "    \n",
    "    tf = os.path.basename(tablefile)\n",
    "\n",
    "    # load the data, create a column with CHROM-POS for locusID\n",
    "    df = pandas.read_csv(tablefile, sep='\\t')\n",
    "    print(f'{tf} has {len(df.index)} rows (includes multiallelic)')\n",
    "    df['locus'] = [\"%s-%s\" % (contig, pos) for (contig, pos) in zip(df['CHROM'].tolist(), df['POS'].tolist())]\n",
    "    df = get_varscan_names(df, tablefile)\n",
    "    \n",
    "    # keep only columns for this variety\n",
    "    cols = [col for col in df.columns if '.' not in col or col.split(\".\")[0] in varlist[variety]]  ### added\n",
    "    df = df[[col for col in df.columns if col in cols]].copy()                                     ### added\n",
    "    \n",
    "    return df, tf\n",
    "dview['load_data'] = load_data\n",
    "\n",
    "def write_file(tablefile, df, tipe, variety):\n",
    "    import pandas\n",
    "    import os\n",
    "    \"\"\"Write filtered pandas.dataframe to file using args to create file name.\"\"\"\n",
    "#     newfile = tablefile.replace(\".txt\", f\"_{tipe}.txt\")                   ########## commented out\n",
    "    write_dir = os.path.dirname(tablefile) + f\"_{variety}\"                  ########## added\n",
    "    bname = os.path.basename(tablefile).replace(\".txt\", f\"_{tipe}_{variety}.txt\")##### added   \n",
    "    newfile = os.path.join(write_dir, bname)                              ########## added\n",
    "    print(f'{tipe}_path = ', newfile)                                     ########## added\n",
    "    \n",
    "    df.to_csv(newfile, index=False, sep='\\t')\n",
    "    print('finished filtering VariantsToTable file: %s' % newfile)\n",
    "dview['write_file'] = write_file\n",
    "\n",
    "def adjust_freqs(smalldf):\n",
    "    \"\"\"\n",
    "    For loci with REF=N, set freqs of pools with REF=N in GT to numpy.nan.\n",
    "    Set alt freqs with respect to the second alt allele.\n",
    "    \n",
    "    Positional arguments:\n",
    "    smalldf - pandas.dataframe; df with only REF=N\n",
    "    \n",
    "    Returns:\n",
    "    ndf - smalldf with adjusted freqs in zeroth row\n",
    "    \"\"\"\n",
    "    import pandas\n",
    "    import numpy\n",
    "    gtcols = [col for col in smalldf.columns if 'GT' in col]\n",
    "\n",
    "    for col in gtcols:\n",
    "        gt = smalldf.loc[1, col]\n",
    "        if isinstance(gt, str):\n",
    "            freqcol = col.split(\".\")[0] + '.FREQ'\n",
    "            if not gt == 'N/N':\n",
    "                freq = smalldf.loc[0, freqcol]\n",
    "                if isinstance(freq, str):\n",
    "                    if \"%\" in freq:\n",
    "                        newfreq = \"%s%%\" % (100 - float(freq.split(\"%\")[0]))\n",
    "                        smalldf.loc[0, freqcol] = newfreq\n",
    "            else:\n",
    "                # if gt = N/N, adjust to undefined\n",
    "                smalldf.loc[1, freqcol] = numpy.nan\n",
    "        gt2 = smalldf.loc[0, col]\n",
    "        if isinstance(gt2, str):\n",
    "            if gt == 'N/N':\n",
    "                # if gt = N/N, adjust to undefined\n",
    "                smalldf.loc[0, freqcol] = numpy.nan\n",
    "    return smalldf\n",
    "dview['adjust_freqs'] = adjust_freqs\n",
    "\n",
    "def get_refn_snps(df, tipe, ndfs=None):\n",
    "    \"\"\"\n",
    "    Isolate polymorphisms with REF=N but two ALT single nuleodite alleles.\n",
    "    \n",
    "    Positional arguments:\n",
    "    df - pandas.dataframe; current filtered VariantsToTable output\n",
    "    \n",
    "    Returns:\n",
    "    dfs - list of loci (pandas.dataframes) with REF=N and two ALT alleles, counts with respect to second ALT\n",
    "    ndfs - return from pandas.conat(dfs)\n",
    "    \"\"\"\n",
    "    import pandas\n",
    "    # as far as I can tell, crisp output from convert_pooled_vcf.py will not output REF = N\n",
    "    ndf = df[df['REF'] == 'N'].copy()\n",
    "    ndf = ndf[ndf['TYPE'] == tipe].copy()\n",
    "    ncount = table(ndf['locus'])\n",
    "    nloci = [locus for locus in ncount if ncount[locus] == 2]\n",
    "    ndf = ndf[ndf['locus'].isin(nloci)].copy()\n",
    "    dfs = []\n",
    "    for locus in uni(ndf['locus']):\n",
    "        smalldf = ndf[ndf['locus'] == locus].copy()\n",
    "        if len(smalldf.index) == 2:\n",
    "            smalldf.index = range(len(smalldf.index))\n",
    "            smalldf = adjust_freqs(smalldf)\n",
    "            smalldf.loc[0,'ALT'] = \"%s+%s\" % (smalldf.loc[0,'ALT'], smalldf.loc[1,\"ALT\"])\n",
    "            dfs.append(pandas.DataFrame(smalldf.loc[0,:]).T)\n",
    "    if len(dfs) > 0:\n",
    "        ndfs = pandas.concat(dfs)\n",
    "    return (dfs, ndfs)\n",
    "dview['get_refn_snps'] = get_refn_snps\n",
    "\n",
    "def keep_snps(df, tf):\n",
    "    \"\"\"\n",
    "    Count CHROM-POS (locus) and keep only those with one ALT.\n",
    "    \n",
    "    Positional arguments:\n",
    "    df - pandas.dataframe; currently filtered VariantsToTable output\n",
    "    tf - basename of path to VariantsToTable output\n",
    "    Returns:\n",
    "    df - pandas.dataframe; non-multiallelic-filtered VariantsToTable output\n",
    "    \"\"\"\n",
    "    import pandas\n",
    "    loccount = table(df['locus'])\n",
    "    goodloci = [locus for locus in loccount if loccount[locus] == 1]\n",
    "    print(f'{tf} has {len(goodloci)} good loci (non-multiallelic)')\n",
    "\n",
    "    # filter df for multiallelic (multiple lines), REF != N\n",
    "    df = df[df['locus'].isin(goodloci)].copy()\n",
    "    df = df[df['REF'] != 'N'].copy()\n",
    "    return df\n",
    "dview['keep_snps'] = keep_snps\n",
    "\n",
    "def filter_missing_data(df, tf, tipe):\n",
    "    \"\"\"\n",
    "    Remove loci with < 25% missing data.\n",
    "    Count numpy.nan in .FREQ col to assess % missing data.\n",
    "    \n",
    "    Positional arguments:\n",
    "    df - pandas.dataframe; VariantsToTable output\n",
    "    tf - str; basename of tablefile\n",
    "    tipe - str; one of either \"SNP\" or \"INDEL\"\n",
    "    \n",
    "    Returns:\n",
    "    df - pandas.dataframe; missing data-filtered VariantsToTable output\n",
    "    \"\"\"\n",
    "    import tqdm\n",
    "    import pandas\n",
    "    import math\n",
    "    freqcols = [col for col in df.columns if '.FREQ' in col]\n",
    "    copy = get_copy(df, freqcols)\n",
    "    keepers = []\n",
    "    # else statement for running single pos.path.(megagamtos.path.yte) through:\n",
    "    thresh = math.floor(0.25 * len(freqcols)) if len(freqcols) > 1 else 1\n",
    "    for locus in tqdm.tqdm(copy.columns):\n",
    "        # if there is less than 25% missing data:\n",
    "        # the only time x != x is when x is nan (fastest way to count it)\n",
    "        count = sum(1 for x in copy[locus] if x != x)\n",
    "        if count < thresh:\n",
    "            keepers.append(locus)\n",
    "    df = df[df.index.isin(keepers)].copy()\n",
    "    df.index = range(len(df.index))\n",
    "    return df\n",
    "dview['filter_missing_data'] = filter_missing_data\n",
    "\n",
    "def get_copy(df, cols):\n",
    "    \"\"\"\n",
    "    Transpose dataframe using specific columns (that will be index after transformation).\n",
    "    Doing so helps speed things up.\n",
    "    \"\"\"\n",
    "    import pandas\n",
    "    return df[cols].T.copy()\n",
    "dview['get_copy'] = get_copy\n",
    "\n",
    "def get_variety_freq_cutoffs(variety, ploidy):\n",
    "    \"\"\"\n",
    "    Use number of pops per variety to determine lowfreq, highfreq.\n",
    "    Differs from pipeline.\n",
    "    \"\"\"\n",
    "    lowfreq = 1/sum([popploidy for pop,popploidy in ploidy.items() if pop in varlist[variety]])\n",
    "    ###############                                               ##### note diffs with get_freq_cutoffs(tablefile)\n",
    "    highfreq = 1 - lowfreq\n",
    "    return lowfreq, highfreq\n",
    "dview['get_variety_freq_cutoffs'] = get_variety_freq_cutoffs\n",
    "\n",
    "def filter_freq(df, tf, tipe, tablefile, variety):\n",
    "    \"\"\"\n",
    "    Keep fixed loci.\n",
    "    \n",
    "    Positional arguments:\n",
    "    df - pandas.dataframe; VariantsToTable output\n",
    "    tablefile - path to VariantsToTable output - used to find ploidy etc\n",
    "    tf - str; basename of tablefile\n",
    "    tipe - str; one of either \"SNP\" or \"INDEL\"\n",
    "    \n",
    "    Returns:\n",
    "    df - pandas.dataframe; freq-filtered VariantsToTable output\n",
    "    \"\"\"\n",
    "    import tqdm\n",
    "    import pandas\n",
    "    import os\n",
    "    import math\n",
    "    # believe it or not, it's faster to do qual and freq filtering in two steps vs an 'and' statement\n",
    "#     lowfreq, highfreq = get_freq_cutoffs(tablefile)                                         ############ removed\n",
    "#     print(f'filtering for global frequency ({lowfreq}, {highfreq})...')                     ############ moved\n",
    "    df.reset_index(drop=True, inplace=True)\n",
    "    \n",
    "    # prep for filtering\n",
    "    freqcols = [col for col in df.columns if '.FREQ' in col]\n",
    "    pool = os.path.basename(os.path.dirname(os.path.dirname(os.path.dirname(tablefile))))     ############ changed\n",
    "    parentdir = os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(tablefile)))) ############\n",
    "    ploidy = pklload(os.path.join(parentdir, f'{pool}/pkl_files/ploidy.pkl'))[pool]           ############\n",
    "    lowfreq, highfreq = get_variety_freq_cutoffs(variety, ploidy)                             ############ added\n",
    "    print(f'filtering for global frequency ({lowfreq}, {highfreq})...')                       ## moved from above\n",
    "    \n",
    "    # carry on with poolseq datas\n",
    "    filtloci = []\n",
    "    afs = []\n",
    "    copy = get_copy(df, freqcols)\n",
    "    for locus in tqdm.tqdm(copy.columns):\n",
    "        freqs = dict((samp.replace(\".FREQ\",\"\"),freq) for (samp,freq)\n",
    "                     in copy[locus].str.rstrip('%').astype('float').items()\n",
    "                     if not math.isnan(freq))  # faster than .str.rstrip('%').astype('float').dropna()\n",
    "        if len(freqs) > 0:  # avoid loci with all freqs masked (avoid ZeroDivisionError)\n",
    "            # calc globfreq using the samps/ploidy that are present for this locus\n",
    "            globfreq = sum([ploidy[samp]*(freq/100)\n",
    "                            for (samp,freq) in freqs.items()]) / sum([ploidy[samp] for samp in freqs])\n",
    "            if lowfreq <= globfreq <= highfreq:\n",
    "                filtloci.append(locus)\n",
    "                # since we're going in order of rows in df ...\n",
    "                # ... we can use afs to replace AF col later since we reduce df to filtloci\n",
    "                afs.append(globfreq)\n",
    "                # which is about 40x faster than: df.loc[locus, 'AF'] = globfreq\n",
    "    print(f'{tf} has {len(filtloci)} {tipe}s that have global MAF > {lowfreq*100}%')\n",
    "    df = df[df.index.isin(filtloci)].copy()\n",
    "    df.index = range(len(df.index))\n",
    "    df['AF'] = afs\n",
    "    return df\n",
    "dview['filter_freq'] = filter_freq\n",
    "\n",
    "def filter_qual(df, tf, tipe, tablefile, variety):\n",
    "    \"\"\"\n",
    "    mask freqs that have GQ < 20.\n",
    "    \n",
    "    Positional arguments:\n",
    "    df - pandas.dataframe; VariantsToTable output\n",
    "    tf - str; basename of tablefile\n",
    "    tipe - str; one of either \"SNP\" or \"INDEL\"\n",
    "    \n",
    "    Returns: pandas.dataframe; quality-filtered VariantsToTable output\n",
    "    - FREQ and GT are masked (numpy.nan) if GQ < 20\n",
    "    \"\"\"\n",
    "    import tqdm\n",
    "    import pandas\n",
    "    import numpy\n",
    "    gqcols = [col for col in df.columns if '.GQ' in col]\n",
    "    print(f'masking bad freqs for {len(gqcols)} pools...')\n",
    "    for col in tqdm.tqdm(gqcols):\n",
    "        freqcol = col.replace(\".GQ\", \".FREQ\")\n",
    "#         gtcol = col.replace(\".GQ\", \".GT\")  # pretty sure this is depricated\n",
    "        # badloci True if qual < 20\n",
    "#         df.loc[df[col] < 20, [freqcol, gtcol]] = np.nan\n",
    "        df.loc[df[col] < 20, freqcol] = numpy.nan\n",
    "\n",
    "    print('filtering for missing data ...')\n",
    "    df = filter_missing_data(df, tf, tipe)\n",
    "\n",
    "    if len(df.index) > 0:\n",
    "        print(f'{tf} has {len(df.index)} {tipe}s that have GQ >= 20 and < 25% missing data')\n",
    "        df = filter_freq(df, tf, tipe, tablefile, variety)\n",
    "        df.index = range(len(df.index))\n",
    "    else:\n",
    "        print(f'{tf} did not have any {tipe}s that have GQ >= 20 for >= 75% of pops' +\n",
    "              '\\nnot bothering to filter for freq')\n",
    "#         df = drop_freq_cols(df)\n",
    "    return df\n",
    "dview['filter_qual'] = filter_qual\n",
    "\n",
    "\n",
    "def main(tablefile, tipe='SNP', parentdir=None, ret=True, variety=None):   ########## changed default args\n",
    "    import sys\n",
    "    import pandas\n",
    "    import numpy\n",
    "    import math\n",
    "    import tqdm\n",
    "    import os\n",
    "    from collections import Counter\n",
    "    # load the data\n",
    "    df, tf = load_data(tablefile, variety)\n",
    "    \n",
    "    # filter only SNPs\n",
    "    df = df[df['TYPE'] == tipe].copy()\n",
    "\n",
    "    # determine loci with REF=N but biallelic otherwise\n",
    "    if tipe == 'SNP':\n",
    "        dfs, ndfs = get_refn_snps(df, tipe)\n",
    "\n",
    "        # determine which loci are multiallelic\n",
    "        df = keep_snps(df, tf)\n",
    "    \n",
    "    if len(df.index) == 0:\n",
    "        if ret is True:\n",
    "            return df\n",
    "        else:\n",
    "            # save\n",
    "            write_file(tablefile, df, tipe)\n",
    "\n",
    "    # add in loci with REF=N but biallelic otherwise\n",
    "    if tipe == 'SNP' and len(dfs) > 0:\n",
    "        print(f'{tf} has {len(ndfs.index)} biallelic {tipe}s with REF=N')\n",
    "        dfs.append(df)\n",
    "        df = pandas.concat(dfs)\n",
    "\n",
    "    # filter for quality and missing data\n",
    "    df.index = range(len(df.index))\n",
    "    if 'varscan' in tf and tipe == 'SNP':\n",
    "        # if we allow to continue for INDEL, each line is treated as a locus (not true for INDEL)\n",
    "        df = filter_qual(df, tf, tipe, tablefile, variety)\n",
    "\n",
    "\n",
    "########################################################################################################\n",
    "    # look for filtering options called at 00_start.py\n",
    "    if parentdir is not None and tipe == 'SNP':\n",
    "        # translate stitched (if called at 00_start)   ############## no need to translate for DF\n",
    "\n",
    "        # remove repeats (if called at 00_start) - want to remove repeats before paralogs\n",
    "        df = remove_repeats(df.copy(),\n",
    "                            parentdir,\n",
    "                            tablefile,\n",
    "                            os.path.basename(os.path.dirname(os.path.dirname(os.path.dirname(tablefile)))), ## added\n",
    "                            variety)  ###### added\n",
    "#                             op.basename(pooldir))  # commented out\n",
    "\n",
    "        # remove paralog SNPs (if called at 00_start)\n",
    "        df = remove_paralogs(df.copy(), parentdir, tablefile,\n",
    "                             os.path.basename(os.path.dirname(os.path.dirname(os.path.dirname(tablefile)))),## added\n",
    "                             variety) ###### added\n",
    "########################################################################################################\n",
    "\n",
    "        \n",
    "    if ret is True:\n",
    "        print('returning df')\n",
    "        return df\n",
    "    else:\n",
    "        # save\n",
    "        write_file(tablefile, df, tipe, variety)\n",
    "\n",
    "dview['main'] = main\n",
    "dview['uni'] = uni\n",
    "from pythonimports import table # in case I use 'table' in an iteration\n",
    "dview['table'] = table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "metadata": {},
   "outputs": [],
   "source": [
    "def remove_paralogs(snps, parentdir, snpspath, pool, variety):\n",
    "    \"\"\"\n",
    "    Remove sites from snptable that are thought to have multiple gene copies align to this position.\n",
    "    \n",
    "    # assumes\n",
    "    # paralog file has 'CHROM' and 'locus' in the header (best if this is the only data, reads in quicker)\n",
    "    #   where CHROM is the reference chromosome/scaffold\n",
    "    #   where locus is hyphen-separated CHROM-POS\n",
    "    \n",
    "    # paralog file is created from calling SNPs on haplotype data as diploid\n",
    "    #   no need to worry about translating stiched -> unstitched if SNPs called on same reference.\n",
    "    \"\"\"\n",
    "    import os, pandas\n",
    "    parpkl = os.path.join(parentdir, f'{pool}/pkl_files/paralog_snps.pkl')\n",
    "    if os.path.exists(parpkl):\n",
    "        # read in paralogfile\n",
    "#         paralogdict = pklload(parpkl)                                            ############ commented out\n",
    "#         if paralogdict[pool] is not None:                                        ############ commented out\n",
    "        if True:                                                                   ########## added\n",
    "            print('Removing paralogs sites ...')\n",
    "#             paralogs = pd.read_csv(paralogdict[pool], sep='\\t')                  ############ commented out\n",
    "            refdir = '/data/database/DouglasFir_ref_genome'                        ############ added\n",
    "            paralogfile = os.path.join(refdir, 'DF_mega-varscan_all_bedfiles_SNP_paralog_snps.txt')# added\n",
    "            paralogs = pandas.read_table(paralogfile)                                  ############ added\n",
    "            # remove and isolate paralogs from snps\n",
    "            truths = snps['locus'].isin(paralogs['locus'])\n",
    "            found_paralogs = snps[truths].copy()\n",
    "            snps = snps[~truths].copy()\n",
    "            snps.index = range(len(snps.index))\n",
    "\n",
    "            # write paralogs to a file\n",
    "#             parafile = snpspath.replace(\".txt\", \"_PARALOGS.txt\")                 ########## commented out\n",
    "            \n",
    "            write_dir = os.path.dirname(snpspath) + f\"_{variety}\"                  ########## added\n",
    "            bname = os.path.basename(snpspath).replace(\".txt\", f\"_PARALOGS_{variety}.txt\")### added   \n",
    "            parafile = os.path.join(write_dir, bname)                              ########## added\n",
    "            print('paralog_path = ', parafile)                                     ########## added\n",
    "            \n",
    "            found_paralogs.to_csv(parafile, sep='\\t', index=False)\n",
    "            print(f'{os.path.basename(snpspath)} has {len(snps.index)} non-paralog SNPs')\n",
    "    return snps\n",
    "dview['remove_paralogs'] = remove_paralogs\n",
    "\n",
    "\n",
    "def remove_repeats(snps, parentdir, snpspath, pool, variety):\n",
    "    \"\"\"\n",
    "    Remove SNPs that are found to be in repeat-masked regions.\n",
    "    \n",
    "    # assumes\n",
    "    # that the positions have been translated BEFORE removing repeats\n",
    "        # took forever to create unstitched repeat regions, don't want to translate repeat file\n",
    "        # this way I can just use unstitched chrom if reference is stitched\n",
    "    # repeat file has a header ('CHROM', 'start', 'stop')\n",
    "    # start and stop positions of repeat regions are 1-based\n",
    "    \"\"\"\n",
    "    import pandas\n",
    "    import tqdm\n",
    "    import os\n",
    "    reppkl = os.path.join(parentdir, f'{pool}/pkl_files/repeat_regions.pkl')\n",
    "    if os.path.exists(reppkl):\n",
    "        # read in repeat regions\n",
    "#         repeatdict = pklload(reppkl)                                             ########## commented out\n",
    "#         if repeatdict[pool] is not None:                                         ########## commented out\n",
    "        if True:                                                                   ########## added\n",
    "            print('Removing repeat regions ...')\n",
    "            # if user selected translation be applied to this pool\n",
    "#             repeats = pd.read_csv(repeatdict[pool], sep='\\t')                    ########## commented out\n",
    "            repeats = pandas.read_table('/data/database/DouglasFir_ref_genome/DF_ref_edit_repeats.txt')   #### added\n",
    "            # figure out if data is from stitched or not\n",
    "            if 'unstitched_chrom' in snps.columns:\n",
    "                # then the snps have been translated: stitched -> unstitched\n",
    "                chromcol = 'unstitched_chrom'\n",
    "                poscol = 'unstitched_pos'\n",
    "                print('\\tsnps have been translated')\n",
    "            else:\n",
    "                # otherwise SNPs were called on unstitched reference\n",
    "                chromcol = 'CHROM'\n",
    "                poscol = 'POS'\n",
    "                print('\\tsnps have not been translated')\n",
    "            # reduce repeats to the chroms that matter (helps speed up lookups)\n",
    "            repeats = repeats[repeats['CHROM'].isin(snps[chromcol].tolist())].copy()\n",
    "\n",
    "            # isolate SNPs in repeat regions\n",
    "            repeat_snps = []\n",
    "            for chrom in tqdm.tqdm(uni(snps[chromcol])):\n",
    "                reps = repeats[repeats['CHROM'] == chrom].copy()\n",
    "                mysnps = snps[snps[chromcol] == chrom].copy()\n",
    "                if len(reps.index) > 0 and len(mysnps.index) > 0:\n",
    "                    for row in mysnps.index:\n",
    "                        pos = snps.loc[row, poscol]  # index is maintained from snps to mysnsps\n",
    "                        df = reps[reps['stop'].astype(int) >= int(pos)].copy()\n",
    "                        df = df[df['start'].astype(int) <= int(pos)].copy()\n",
    "                        if len(df.index) > 0:\n",
    "                            assert len(df.index) == 1\n",
    "                            repeat_snps.append(row)\n",
    "\n",
    "            # save repeats\n",
    "            print(f'\\tSaving {len(repeat_snps)} repeat regions')\n",
    "#             repeat_path = snpspath.replace(\".txt\", \"_REPEATS.txt\")               ########## comm ented out\n",
    "            write_dir = os.path.dirname(snpspath) + f\"_{variety}\"                  ########## added\n",
    "            bname = os.path.basename(snpspath).replace(\".txt\", f\"_REPEATS_{variety}.txt\")### added   \n",
    "            repeat_path = os.path.join(write_dir, bname)                           ########## added\n",
    "            print('repeat_path = ', repeat_path)                                   ########## added\n",
    "            \n",
    "            myrepeats = snps[snps.index.isin(repeat_snps)].copy()\n",
    "            myrepeats.to_csv(repeat_path, sep='\\t', index=False)\n",
    "\n",
    "            # remove SNPs in repeat regions\n",
    "            snps = snps[~snps.index.isin(repeat_snps)].copy()\n",
    "            snps.index = range(len(snps.index))\n",
    "\n",
    "            print(f'{os.path.basename(snpspath)} has {len(snps.index)} SNPs outside of repeat regions')\n",
    "\n",
    "    return snps\n",
    "dview['remove_repeats'] = remove_repeats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# envdata has variety ID\n",
    "envdata = pd.read_table('/data/projects/pool_seq/environemental_data/df_std_env-19variables.txt')\n",
    "envdata = envdata[envdata['our_id']==envdata['our_id']]  # removes irrelevant pop with our_id=nan\n",
    "\n",
    "pool2var = {}\n",
    "varlist = {}\n",
    "for row in envdata.index:\n",
    "    pool = envdata.loc[row, 'our_id']\n",
    "    variety = envdata.loc[row, 'Variety']\n",
    "    pool2var[pool] = variety\n",
    "    if variety not in varlist:\n",
    "        varlist[variety] = []\n",
    "    varlist[variety].append(pool)\n",
    "dview['varlist'] = varlist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FDC\n",
      "FDI\n"
     ]
    }
   ],
   "source": [
    "# create directories to save files\n",
    "for variety in varlist.keys():\n",
    "    makedir(f'/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/01_unfiltered_{variety}')\n",
    "    print(variety)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  0%|          | 0/45 [00:00<?, ?it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DF_pooled_varscan_bedfile_0391_table.txt has 11890 rows (includes multiallelic)\n",
      "renaming varscan columns ...\n",
      "DF_pooled_varscan_bedfile_0391_table.txt has 10219 good loci (non-multiallelic)\n",
      "masking bad freqs for 45 pools...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 45/45 [00:00<00:00, 643.56it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "filtering for missing data ...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 10216/10216 [00:00<00:00, 15204.98it/s]\n",
      "  4%|▍         | 199/4571 [00:00<00:02, 1984.03it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DF_pooled_varscan_bedfile_0391_table.txt has 4571 SNPs that have GQ >= 20 and < 25% missing data\n",
      "filtering for global frequency (0.0002796420581655481, 0.9997203579418344)...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 4571/4571 [00:02<00:00, 2018.31it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DF_pooled_varscan_bedfile_0391_table.txt has 4137 SNPs that have global MAF > 0.02796420581655481%\n",
      "Removing repeat regions ...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "  0%|          | 0/29 [00:00<?, ?it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tsnps have not been translated\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 29/29 [00:05<00:00,  5.19it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tSaving 56 repeat regions\n",
      "repeat_path =  /data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/01_unfiltered_FDC/DF_pooled_varscan_bedfile_0391_table_REPEATS_FDC.txt\n",
      "DF_pooled_varscan_bedfile_0391_table.txt has 4081 SNPs outside of repeat regions\n",
      "Removing paralogs sites ...\n",
      "paralog_path =  /data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/01_unfiltered_FDC/DF_pooled_varscan_bedfile_0391_table_PARALOGS_FDC.txt\n",
      "DF_pooled_varscan_bedfile_0391_table.txt has 4076 non-paralog SNPs\n",
      "SNP_path =  /data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/01_unfiltered_FDC/DF_pooled_varscan_bedfile_0391_table_SNP_FDC.txt\n",
      "finished filtering VariantsToTable file: /data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/01_unfiltered_FDC/DF_pooled_varscan_bedfile_0391_table_SNP_FDC.txt\n"
     ]
    }
   ],
   "source": [
    "# test out filtering\n",
    "tablefile = '/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/01_unfiltered/DF_pooled_varscan_bedfile_0391_table.txt'\n",
    "df = main(tablefile, tipe='SNP', parentdir=\"/data/projects/pool_seq/DF_datasets/DF_pooled_GEA\",\n",
    "          ret=False, variety='FDC')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### now do in parallel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "932"
      ]
     },
     "execution_count": 192,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get all of the varscan outputs\n",
    "files = fs('/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/01_unfiltered',\n",
    "           endswith='table.txt')\n",
    "len(files)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 241,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "932\n",
      "932\n"
     ]
    }
   ],
   "source": [
    "# filter FDC variety\n",
    "jobs = []\n",
    "for f in files:\n",
    "    jobs.append(lview.apply_async(main, f, **{'tipe':'SNP',\n",
    "                                              'parentdir':\"/data/projects/pool_seq/DF_datasets/DF_pooled_GEA\",\n",
    "                                              'ret':False,\n",
    "                                              'variety':'FDC'}))\n",
    "watch_async(jobs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 248,
   "metadata": {},
   "outputs": [],
   "source": [
    "# make sure no errors\n",
    "for j in jobs:\n",
    "    x = j.r"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 249,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "932\n",
      "932\n"
     ]
    }
   ],
   "source": [
    "# filter FDI variety\n",
    "fdijobs = []\n",
    "for f in files:\n",
    "    fdijobs.append(lview.apply_async(main, f, **{'tipe':'SNP',\n",
    "                                              'parentdir':\"/data/projects/pool_seq/DF_datasets/DF_pooled_GEA\",\n",
    "                                              'ret':False,\n",
    "                                              'variety':'FDI'}))\n",
    "watch_async(fdijobs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 250,
   "metadata": {},
   "outputs": [],
   "source": [
    "# make sure no errors\n",
    "for j in fdijobs:\n",
    "    x = j.r"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "metadata": {},
   "outputs": [],
   "source": [
    "# check to see how many files were produced per bedfile\n",
    "# filtfiles = fs('/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/01_unfiltered_FDC',\n",
    "#                endswith='.txt')\n",
    "filtfiles = fs('/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/01_unfiltered_FDI',\n",
    "               endswith='.txt')\n",
    "bedtofiles = {}\n",
    "for f in filtfiles:\n",
    "    bed = f.split(\"bedfile_\")[1].split(\"_table\")[0]\n",
    "    assert float(bed) == int(bed)\n",
    "    if bed not in bedtofiles:\n",
    "        bedtofiles[bed] = []\n",
    "    bedtofiles[bed].append(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 255,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "missing = []\n",
    "for i in range(int(max(bedtofiles.keys()))):\n",
    "    bed = str(i).zfill(4)\n",
    "    if not bed in bedtofiles.keys():\n",
    "        missing.append(files[i])\n",
    "    elif len(bedtofiles[bed]) != 3:\n",
    "        missing.append(files[i])\n",
    "len(missing)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### combine dataframes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pythonimports import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 257,
   "metadata": {},
   "outputs": [],
   "source": [
    "# make new dirs\n",
    "for variety in varlist:\n",
    "    makedir(f'/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/02_baseline_filtered_{variety}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "56 56\n"
     ]
    }
   ],
   "source": [
    "lview,dview = get_client()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_df(f):\n",
    "    import pandas\n",
    "    return pandas.read_table(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "932\n",
      "932\n",
      "FDI REPEATS 932 282344\n"
     ]
    }
   ],
   "source": [
    "# combine dfs\n",
    "for variety in varlist:\n",
    "    d = f'/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/01_unfiltered_{variety}'\n",
    "    dstdir = f'/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/02_baseline_filtered_{variety}'\n",
    "    assert op.exists(dstdir)\n",
    "    for tipe in ['SNP', 'PARALOGS', 'REPEATS']:\n",
    "        files = fs(d, pattern=tipe, endswith='.txt')\n",
    "        jobs = make_jobs(files, read_df, lview)\n",
    "        watch_async(jobs)\n",
    "        df = pd.concat([j.r for j in jobs])\n",
    "        file = op.join(dstdir, f'DF_pooled-varscan_all_bedfiles_{tipe}_{variety}.txt')\n",
    "        print(variety, tipe, len(files), nrow(df))\n",
    "        df.to_csv(file, sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# filter for MAF\n",
    "\n",
    "filter for MAF >= 0.05 for each variety"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pythonimports import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "56 56\n"
     ]
    }
   ],
   "source": [
    "lview,dview = get_client()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_skipto_df(f, skipto, nrows, cols=None, filter_maf=False, **kwargs):\n",
    "    \"\"\"Retrieve dataframe in parallel so that all rows are captured when iterating.\n",
    "    \n",
    "    f = filename to open\n",
    "    skipto = row number to skip, read rows thereafter\n",
    "    nrows = how many rows to read from f after skipto\n",
    "    \"\"\"\n",
    "    import pandas\n",
    "    \n",
    "    if skipto == 0:\n",
    "        df = pandas.read_table(f, nrows=nrows-1)\n",
    "    else:\n",
    "        df = pandas.read_table(f, skiprows=range(1, skipto), nrows=nrows)\n",
    "    \n",
    "    if cols is not None:\n",
    "        if isinstance(cols, str):\n",
    "            cols = [cols]\n",
    "        df = df[cols].copy()\n",
    "    \n",
    "    if filter_maf is True:\n",
    "        return maf_filter(df, **kwargs)\n",
    "    \n",
    "    return df\n",
    "dview['get_skipto_df'] = get_skipto_df\n",
    "\n",
    "def maf_filter(chunk, maf=0.05, **kwargs):\n",
    "    \"\"\"filter minor allele frequency >= maf, create maf column, return df.\"\"\"\n",
    "    import pandas\n",
    "    import os\n",
    "\n",
    "    # filter for MAF\n",
    "    df = chunk[(chunk['AF'].astype(float) >= maf) & (chunk['AF'].astype(float) <= (1-maf))].copy()\n",
    "    # create MAF column\n",
    "    df['MAF'] = df['AF']\n",
    "    df.loc[df['AF'].astype(float) > 0.5, 'MAF'] = 1 - chunk['AF'][chunk['AF'].astype(float) > 0.5]\n",
    "    assert sum(df['MAF']<maf) == 0\n",
    "    \n",
    "    return df\n",
    "dview['maf_filter'] = maf_filter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FDI SNP 9062679\n",
      "FDI PARALOGS 1859\n",
      "FDI REPEATS 282344\n",
      "FDC SNP 9409666\n",
      "FDC PARALOGS 1831\n",
      "FDC REPEATS 291081\n"
     ]
    }
   ],
   "source": [
    "# get linenums for each variety and each type\n",
    "linenums = {}\n",
    "for variety in ['FDI', 'FDC']:\n",
    "    d = f'/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/02_baseline_filtered_{variety}'\n",
    "    for tipe in ['SNP', 'PARALOGS', 'REPEATS']:\n",
    "        f = op.join(d, f'DF_pooled-varscan_all_bedfiles_{tipe}_{variety}.txt')\n",
    "        out = !wc -l $f\n",
    "        linenums[f] = int(out[0].split()[0])-1\n",
    "        print(variety, tipe, linenums[f])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### first filter interior"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "182\n",
      "182\n"
     ]
    }
   ],
   "source": [
    "# read in SNPs in parallel\n",
    "f = '/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/02_baseline_filtered_FDI/DF_pooled-varscan_all_bedfiles_SNP_FDI.txt'\n",
    "nrows = 50000\n",
    "jobs = []\n",
    "count = 0\n",
    "for skipto in range(0, linenums[f], nrows):\n",
    "    num = str(count).zfill(4)\n",
    "    jobs.append(lview.apply_async(get_skipto_df, *(f, skipto, nrows), **{'filter_maf':True, 'maf':0.05}))\n",
    "    count += 1\n",
    "watch_async(jobs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# check for errors\n",
    "for j in jobs:\n",
    "    x = j.r"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2609583, 351)"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# concat filtered data\n",
    "interior = pd.concat([j.r for j in jobs])\n",
    "interior.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# since I have this SNP file read in already, go to Recalculate RD section, then come back for coastal variety"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/02_baseline_filtered_FDI/DF_pooled-varscan_all_bedfiles_SNP_FDI.txt': 9062679,\n",
       " '/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/02_baseline_filtered_FDI/DF_pooled-varscan_all_bedfiles_PARALOGS_FDI.txt': 1859,\n",
       " '/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/02_baseline_filtered_FDI/DF_pooled-varscan_all_bedfiles_REPEATS_FDI.txt': 282344,\n",
       " '/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/02_baseline_filtered_FDC/DF_pooled-varscan_all_bedfiles_SNP_FDC.txt': 9409666,\n",
       " '/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/02_baseline_filtered_FDC/DF_pooled-varscan_all_bedfiles_PARALOGS_FDC.txt': 1831,\n",
       " '/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/02_baseline_filtered_FDC/DF_pooled-varscan_all_bedfiles_REPEATS_FDC.txt': 291081}"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "linenums"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DF_pooled-varscan_all_bedfiles_PARALOGS_FDI.txt (1859, 350)\n",
      "\t (988, 351)\n",
      "DF_pooled-varscan_all_bedfiles_REPEATS_FDI.txt (282344, 350)\n",
      "\t (89609, 351)\n"
     ]
    }
   ],
   "source": [
    "# now do paralogs and repeats\n",
    "fdi_remaining = {}\n",
    "for key in keys(linenums)[1:3]:\n",
    "    df = pd.read_csv(key, sep='\\t')\n",
    "    print(op.basename(key), df.shape)\n",
    "    fdi_remaining[key] = maf_filter(df)\n",
    "    print('\\t', fdi_remaining[key].shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# go to recalc RD and then come back for coastal variety paralogs and repeats"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### now do coastal variety"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "189\n",
      "189\n"
     ]
    }
   ],
   "source": [
    "# read in SNPs in parallel\n",
    "f = '/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/02_baseline_filtered_FDC/DF_pooled-varscan_all_bedfiles_SNP_FDC.txt'\n",
    "nrows = 50000\n",
    "jobs = []\n",
    "count = 0\n",
    "for skipto in range(0, linenums[f], nrows):\n",
    "    num = str(count).zfill(4)\n",
    "    jobs.append(lview.apply_async(get_skipto_df, *(f, skipto, nrows), **{'filter_maf':True, 'maf':0.05}))\n",
    "    count += 1\n",
    "watch_async(jobs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "# check for errors\n",
    "for j in jobs:\n",
    "    x = j.r"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2350673, 375)"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# concat filtered data\n",
    "coastal = pd.concat([j.r for j in jobs])\n",
    "coastal.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/02_baseline_filtered_FDC/DF_pooled-varscan_all_bedfiles_PARALOGS_FDC.txt',\n",
       " '/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/02_baseline_filtered_FDC/DF_pooled-varscan_all_bedfiles_REPEATS_FDC.txt']"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "keys(linenums)[4:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DF_pooled-varscan_all_bedfiles_PARALOGS_FDC.txt (1831, 374)\n",
      "\t (1110, 375)\n",
      "DF_pooled-varscan_all_bedfiles_REPEATS_FDC.txt (291081, 374)\n",
      "\t (79955, 375)\n"
     ]
    }
   ],
   "source": [
    "# now do paralogs and repeats\n",
    "fdc_remaining = {}\n",
    "for key in keys(linenums)[4:]:\n",
    "    df = pd.read_csv(key, sep='\\t')\n",
    "    print(op.basename(key), df.shape)\n",
    "    fdc_remaining[key] = maf_filter(df)\n",
    "    print('\\t', fdc_remaining[key].shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Recalcuate RD\n",
    "\n",
    "Looking at our testdata (1 poolseq pop vs indSeq of same individuals), AD/DP was consistent with the frequency prediction from GATK. We saw that adjusting FREQ to AD / (AD + RD) decreased concordance between the two datasets. So that we are consistent with respect to uncorrected and corrected, I'm adjusting RD = DP - AD so we don't have to make adjustments in the future"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "def recalc_rd(df):\n",
    "    \"\"\"Recalculate RD so RD = DP - AD.\"\"\"\n",
    "    rdcols = [col for col in df if '.RD' in col]\n",
    "    for col in nb(rdcols):\n",
    "        pop = col.split(\".\")[0]\n",
    "        df[f'{pop}.RD'] = df[f'{pop}.DP'] - df[f'{pop}.AD']\n",
    "    return df\n",
    "\n",
    "def save_file(df, f):\n",
    "    \"\"\"Save file to background using one of the ipcluster engines so I can contiue working.\"\"\"\n",
    "    import pandas\n",
    "    df.to_csv(f, sep='\\t', index=False)\n",
    "    return f"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/03_maf-p05_RD-recalculated_FDI\n",
      "/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/03_maf-p05_RD-recalculated_FDC\n"
     ]
    }
   ],
   "source": [
    "# make dirs\n",
    "newdirs = {}\n",
    "for variety in ['FDI', 'FDC']:\n",
    "    newdirs[variety] = makedir(f'/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/03_maf-p05_RD-recalculated_{variety}')\n",
    "    print(newdirs[variety])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### recalc interior files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AF</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>TYPE</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>ADP</th>\n",
       "      <th>WT</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>NC</th>\n",
       "      <th>DF_p18.GT</th>\n",
       "      <th>DF_p18.GQ</th>\n",
       "      <th>DF_p18.SDP</th>\n",
       "      <th>DF_p18.DP</th>\n",
       "      <th>DF_p18.FREQ</th>\n",
       "      <th>DF_p18.PVAL</th>\n",
       "      <th>DF_p18.AD</th>\n",
       "      <th>DF_p18.RD</th>\n",
       "      <th>DF_p19.GT</th>\n",
       "      <th>DF_p19.GQ</th>\n",
       "      <th>DF_p19.SDP</th>\n",
       "      <th>DF_p19.DP</th>\n",
       "      <th>DF_p19.FREQ</th>\n",
       "      <th>DF_p19.PVAL</th>\n",
       "      <th>DF_p19.AD</th>\n",
       "      <th>DF_p19.RD</th>\n",
       "      <th>DF_p20.GT</th>\n",
       "      <th>DF_p20.GQ</th>\n",
       "      <th>DF_p20.SDP</th>\n",
       "      <th>DF_p20.DP</th>\n",
       "      <th>DF_p20.FREQ</th>\n",
       "      <th>DF_p20.PVAL</th>\n",
       "      <th>DF_p20.AD</th>\n",
       "      <th>DF_p20.RD</th>\n",
       "      <th>DF_p3.GT</th>\n",
       "      <th>DF_p3.GQ</th>\n",
       "      <th>DF_p3.SDP</th>\n",
       "      <th>DF_p3.DP</th>\n",
       "      <th>DF_p3.FREQ</th>\n",
       "      <th>DF_p3.PVAL</th>\n",
       "      <th>DF_p3.AD</th>\n",
       "      <th>DF_p3.RD</th>\n",
       "      <th>DF_p33.GT</th>\n",
       "      <th>DF_p33.GQ</th>\n",
       "      <th>DF_p33.SDP</th>\n",
       "      <th>DF_p33.DP</th>\n",
       "      <th>DF_p33.FREQ</th>\n",
       "      <th>...</th>\n",
       "      <th>DF_p86.GT</th>\n",
       "      <th>DF_p86.GQ</th>\n",
       "      <th>DF_p86.SDP</th>\n",
       "      <th>DF_p86.DP</th>\n",
       "      <th>DF_p86.FREQ</th>\n",
       "      <th>DF_p86.PVAL</th>\n",
       "      <th>DF_p86.AD</th>\n",
       "      <th>DF_p86.RD</th>\n",
       "      <th>DF_p87.GT</th>\n",
       "      <th>DF_p87.GQ</th>\n",
       "      <th>DF_p87.SDP</th>\n",
       "      <th>DF_p87.DP</th>\n",
       "      <th>DF_p87.FREQ</th>\n",
       "      <th>DF_p87.PVAL</th>\n",
       "      <th>DF_p87.AD</th>\n",
       "      <th>DF_p87.RD</th>\n",
       "      <th>DF_p88.GT</th>\n",
       "      <th>DF_p88.GQ</th>\n",
       "      <th>DF_p88.SDP</th>\n",
       "      <th>DF_p88.DP</th>\n",
       "      <th>DF_p88.FREQ</th>\n",
       "      <th>DF_p88.PVAL</th>\n",
       "      <th>DF_p88.AD</th>\n",
       "      <th>DF_p88.RD</th>\n",
       "      <th>DF_p89.GT</th>\n",
       "      <th>DF_p89.GQ</th>\n",
       "      <th>DF_p89.SDP</th>\n",
       "      <th>DF_p89.DP</th>\n",
       "      <th>DF_p89.FREQ</th>\n",
       "      <th>DF_p89.PVAL</th>\n",
       "      <th>DF_p89.AD</th>\n",
       "      <th>DF_p89.RD</th>\n",
       "      <th>DF_p9.GT</th>\n",
       "      <th>DF_p9.GQ</th>\n",
       "      <th>DF_p9.SDP</th>\n",
       "      <th>DF_p9.DP</th>\n",
       "      <th>DF_p9.FREQ</th>\n",
       "      <th>DF_p9.PVAL</th>\n",
       "      <th>DF_p9.AD</th>\n",
       "      <th>DF_p9.RD</th>\n",
       "      <th>DF_p94.GT</th>\n",
       "      <th>DF_p94.GQ</th>\n",
       "      <th>DF_p94.SDP</th>\n",
       "      <th>DF_p94.DP</th>\n",
       "      <th>DF_p94.FREQ</th>\n",
       "      <th>DF_p94.PVAL</th>\n",
       "      <th>DF_p94.AD</th>\n",
       "      <th>DF_p94.RD</th>\n",
       "      <th>locus</th>\n",
       "      <th>MAF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>jcf7190000000000</td>\n",
       "      <td>77602</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.822964</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>SNP</td>\n",
       "      <td>PASS</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td>34</td>\n",
       "      <td>26</td>\n",
       "      <td>21</td>\n",
       "      <td>G/T</td>\n",
       "      <td>84.0</td>\n",
       "      <td>24</td>\n",
       "      <td>24.0</td>\n",
       "      <td>79.17%</td>\n",
       "      <td>3.682600e-09</td>\n",
       "      <td>19.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>T/T</td>\n",
       "      <td>121.0</td>\n",
       "      <td>26</td>\n",
       "      <td>26.0</td>\n",
       "      <td>92.31%</td>\n",
       "      <td>7.622200e-13</td>\n",
       "      <td>24.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T/T</td>\n",
       "      <td>87.0</td>\n",
       "      <td>20</td>\n",
       "      <td>20.0</td>\n",
       "      <td>90%</td>\n",
       "      <td>1.675800e-09</td>\n",
       "      <td>18.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>G/T</td>\n",
       "      <td>29.0</td>\n",
       "      <td>9</td>\n",
       "      <td>9.0</td>\n",
       "      <td>77.78%</td>\n",
       "      <td>1.131200e-03</td>\n",
       "      <td>7.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>T/T</td>\n",
       "      <td>44.0</td>\n",
       "      <td>12</td>\n",
       "      <td>12.0</td>\n",
       "      <td>83.33%</td>\n",
       "      <td>3.365200e-05</td>\n",
       "      <td>10.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>T/T</td>\n",
       "      <td>52.0</td>\n",
       "      <td>10</td>\n",
       "      <td>10.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>5.412500e-06</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>T/T</td>\n",
       "      <td>52.0</td>\n",
       "      <td>15</td>\n",
       "      <td>15.0</td>\n",
       "      <td>80%</td>\n",
       "      <td>5.260500e-06</td>\n",
       "      <td>12.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>T/T</td>\n",
       "      <td>39.0</td>\n",
       "      <td>11</td>\n",
       "      <td>11.0</td>\n",
       "      <td>81.82%</td>\n",
       "      <td>0.000111</td>\n",
       "      <td>9.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>G/T</td>\n",
       "      <td>31.0</td>\n",
       "      <td>12</td>\n",
       "      <td>12.0</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.730400e-04</td>\n",
       "      <td>8.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>jcf7190000000000-77602</td>\n",
       "      <td>0.177036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>jcf7190000000000</td>\n",
       "      <td>77657</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.808995</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>SNP</td>\n",
       "      <td>PASS</td>\n",
       "      <td>16</td>\n",
       "      <td>5</td>\n",
       "      <td>51</td>\n",
       "      <td>26</td>\n",
       "      <td>5</td>\n",
       "      <td>C/C</td>\n",
       "      <td>160.0</td>\n",
       "      <td>43</td>\n",
       "      <td>43.0</td>\n",
       "      <td>81.4%</td>\n",
       "      <td>9.593300e-17</td>\n",
       "      <td>35.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>128.0</td>\n",
       "      <td>34</td>\n",
       "      <td>34.0</td>\n",
       "      <td>82.35%</td>\n",
       "      <td>1.349000e-13</td>\n",
       "      <td>28.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C/C</td>\n",
       "      <td>71.0</td>\n",
       "      <td>20</td>\n",
       "      <td>20.0</td>\n",
       "      <td>80%</td>\n",
       "      <td>7.708600e-08</td>\n",
       "      <td>16.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>47.0</td>\n",
       "      <td>11</td>\n",
       "      <td>11.0</td>\n",
       "      <td>90.91%</td>\n",
       "      <td>...</td>\n",
       "      <td>C/C</td>\n",
       "      <td>68.0</td>\n",
       "      <td>18</td>\n",
       "      <td>18.0</td>\n",
       "      <td>83.33%</td>\n",
       "      <td>1.465500e-07</td>\n",
       "      <td>15.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>81.0</td>\n",
       "      <td>17</td>\n",
       "      <td>17.0</td>\n",
       "      <td>94.12%</td>\n",
       "      <td>7.713400e-09</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>60.0</td>\n",
       "      <td>15</td>\n",
       "      <td>15.0</td>\n",
       "      <td>86.67%</td>\n",
       "      <td>8.767500e-07</td>\n",
       "      <td>13.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>79.0</td>\n",
       "      <td>23</td>\n",
       "      <td>23.0</td>\n",
       "      <td>78.26%</td>\n",
       "      <td>1.193700e-08</td>\n",
       "      <td>18.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>49.0</td>\n",
       "      <td>17</td>\n",
       "      <td>17.0</td>\n",
       "      <td>70.59%</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>12.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>60.0</td>\n",
       "      <td>18</td>\n",
       "      <td>18.0</td>\n",
       "      <td>77.78%</td>\n",
       "      <td>8.060500e-07</td>\n",
       "      <td>14.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>jcf7190000000000-77657</td>\n",
       "      <td>0.191005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>jcf7190000000000</td>\n",
       "      <td>77738</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.804214</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>SNP</td>\n",
       "      <td>PASS</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>49</td>\n",
       "      <td>28</td>\n",
       "      <td>8</td>\n",
       "      <td>T/C</td>\n",
       "      <td>103.0</td>\n",
       "      <td>32</td>\n",
       "      <td>32.0</td>\n",
       "      <td>75%</td>\n",
       "      <td>4.196400e-11</td>\n",
       "      <td>24.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>126.0</td>\n",
       "      <td>27</td>\n",
       "      <td>27.0</td>\n",
       "      <td>92.59%</td>\n",
       "      <td>2.085300e-13</td>\n",
       "      <td>25.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C/C</td>\n",
       "      <td>76.0</td>\n",
       "      <td>18</td>\n",
       "      <td>18.0</td>\n",
       "      <td>88.89%</td>\n",
       "      <td>2.093600e-08</td>\n",
       "      <td>16.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>29.0</td>\n",
       "      <td>9</td>\n",
       "      <td>9.0</td>\n",
       "      <td>77.78%</td>\n",
       "      <td>...</td>\n",
       "      <td>C/C</td>\n",
       "      <td>71.0</td>\n",
       "      <td>18</td>\n",
       "      <td>18.0</td>\n",
       "      <td>88.24%</td>\n",
       "      <td>7.327700e-08</td>\n",
       "      <td>15.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>52.0</td>\n",
       "      <td>10</td>\n",
       "      <td>10.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>5.412500e-06</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C/C</td>\n",
       "      <td>105.0</td>\n",
       "      <td>25</td>\n",
       "      <td>25.0</td>\n",
       "      <td>88%</td>\n",
       "      <td>2.591600e-11</td>\n",
       "      <td>22.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>18.0</td>\n",
       "      <td>9</td>\n",
       "      <td>9.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.014706</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>55.0</td>\n",
       "      <td>14</td>\n",
       "      <td>14.0</td>\n",
       "      <td>85.71%</td>\n",
       "      <td>2.991300e-06</td>\n",
       "      <td>12.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>jcf7190000000000-77738</td>\n",
       "      <td>0.195786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>jcf7190000000000</td>\n",
       "      <td>77764</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.933934</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>SNP</td>\n",
       "      <td>PASS</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>28</td>\n",
       "      <td>50</td>\n",
       "      <td>8</td>\n",
       "      <td>C/C</td>\n",
       "      <td>127.0</td>\n",
       "      <td>29</td>\n",
       "      <td>29.0</td>\n",
       "      <td>89.66%</td>\n",
       "      <td>1.649600e-13</td>\n",
       "      <td>26.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>117.0</td>\n",
       "      <td>21</td>\n",
       "      <td>21.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.857800e-12</td>\n",
       "      <td>21.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C/C</td>\n",
       "      <td>76.0</td>\n",
       "      <td>14</td>\n",
       "      <td>14.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.492700e-08</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>58.0</td>\n",
       "      <td>11</td>\n",
       "      <td>11.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>...</td>\n",
       "      <td>C/C</td>\n",
       "      <td>71.0</td>\n",
       "      <td>17</td>\n",
       "      <td>17.0</td>\n",
       "      <td>88.24%</td>\n",
       "      <td>7.327700e-08</td>\n",
       "      <td>15.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>52.0</td>\n",
       "      <td>10</td>\n",
       "      <td>10.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>5.412500e-06</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>36.0</td>\n",
       "      <td>9</td>\n",
       "      <td>9.0</td>\n",
       "      <td>88.89%</td>\n",
       "      <td>2.056800e-04</td>\n",
       "      <td>8.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>86.0</td>\n",
       "      <td>18</td>\n",
       "      <td>18.0</td>\n",
       "      <td>94.44%</td>\n",
       "      <td>2.093600e-09</td>\n",
       "      <td>17.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>52.0</td>\n",
       "      <td>10</td>\n",
       "      <td>10.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>52.0</td>\n",
       "      <td>10</td>\n",
       "      <td>10.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>5.412500e-06</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>jcf7190000000000-77764</td>\n",
       "      <td>0.066066</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>jcf7190000000000</td>\n",
       "      <td>77784</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.904187</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>SNP</td>\n",
       "      <td>PASS</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>38</td>\n",
       "      <td>17</td>\n",
       "      <td>G/G</td>\n",
       "      <td>107.0</td>\n",
       "      <td>27</td>\n",
       "      <td>27.0</td>\n",
       "      <td>85.19%</td>\n",
       "      <td>1.616100e-11</td>\n",
       "      <td>23.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>G/G</td>\n",
       "      <td>111.0</td>\n",
       "      <td>20</td>\n",
       "      <td>20.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>7.254400e-12</td>\n",
       "      <td>20.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G/G</td>\n",
       "      <td>64.0</td>\n",
       "      <td>12</td>\n",
       "      <td>12.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.698000e-07</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G/G</td>\n",
       "      <td>46.0</td>\n",
       "      <td>9</td>\n",
       "      <td>9.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>...</td>\n",
       "      <td>G/G</td>\n",
       "      <td>49.0</td>\n",
       "      <td>13</td>\n",
       "      <td>13.0</td>\n",
       "      <td>84.62%</td>\n",
       "      <td>1.009600e-05</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>G/G</td>\n",
       "      <td>41.0</td>\n",
       "      <td>8</td>\n",
       "      <td>8.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>7.770000e-05</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G/G</td>\n",
       "      <td>46.0</td>\n",
       "      <td>9</td>\n",
       "      <td>9.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.056800e-05</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G/G</td>\n",
       "      <td>34.0</td>\n",
       "      <td>10</td>\n",
       "      <td>10.0</td>\n",
       "      <td>80%</td>\n",
       "      <td>3.572300e-04</td>\n",
       "      <td>8.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>G/G</td>\n",
       "      <td>46.0</td>\n",
       "      <td>9</td>\n",
       "      <td>9.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>0.000021</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>37.0</td>\n",
       "      <td>12</td>\n",
       "      <td>12.0</td>\n",
       "      <td>75%</td>\n",
       "      <td>1.682600e-04</td>\n",
       "      <td>9.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>jcf7190000000000-77784</td>\n",
       "      <td>0.095813</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 351 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              CHROM    POS REF ALT        AF  QUAL TYPE FILTER  ADP  WT  HET  \\\n",
       "0  jcf7190000000000  77602   G   T  0.822964 -10.0  SNP   PASS   11   6   34   \n",
       "2  jcf7190000000000  77657   T   C  0.808995 -10.0  SNP   PASS   16   5   51   \n",
       "5  jcf7190000000000  77738   T   C  0.804214 -10.0  SNP   PASS   13   2   49   \n",
       "6  jcf7190000000000  77764   T   C  0.933934 -10.0  SNP   PASS   12   1   28   \n",
       "7  jcf7190000000000  77784   A   G  0.904187 -10.0  SNP   PASS   11   0   32   \n",
       "\n",
       "   HOM  NC DF_p18.GT  DF_p18.GQ  DF_p18.SDP  DF_p18.DP DF_p18.FREQ  \\\n",
       "0   26  21       G/T       84.0          24       24.0      79.17%   \n",
       "2   26   5       C/C      160.0          43       43.0       81.4%   \n",
       "5   28   8       T/C      103.0          32       32.0         75%   \n",
       "6   50   8       C/C      127.0          29       29.0      89.66%   \n",
       "7   38  17       G/G      107.0          27       27.0      85.19%   \n",
       "\n",
       "    DF_p18.PVAL  DF_p18.AD  DF_p18.RD DF_p19.GT  DF_p19.GQ  DF_p19.SDP  \\\n",
       "0  3.682600e-09       19.0        5.0       T/T      121.0          26   \n",
       "2  9.593300e-17       35.0        8.0       C/C      128.0          34   \n",
       "5  4.196400e-11       24.0        8.0       C/C      126.0          27   \n",
       "6  1.649600e-13       26.0        3.0       C/C      117.0          21   \n",
       "7  1.616100e-11       23.0        4.0       G/G      111.0          20   \n",
       "\n",
       "   DF_p19.DP DF_p19.FREQ   DF_p19.PVAL  DF_p19.AD  DF_p19.RD DF_p20.GT  \\\n",
       "0       26.0      92.31%  7.622200e-13       24.0        2.0       ./.   \n",
       "2       34.0      82.35%  1.349000e-13       28.0        6.0       ./.   \n",
       "5       27.0      92.59%  2.085300e-13       25.0        2.0       ./.   \n",
       "6       21.0        100%  1.857800e-12       21.0        0.0       ./.   \n",
       "7       20.0        100%  7.254400e-12       20.0        0.0       ./.   \n",
       "\n",
       "   DF_p20.GQ  DF_p20.SDP  DF_p20.DP DF_p20.FREQ  DF_p20.PVAL  DF_p20.AD  \\\n",
       "0        NaN           2        NaN         NaN          NaN        NaN   \n",
       "2        NaN           3        NaN         NaN          NaN        NaN   \n",
       "5        NaN           3        NaN         NaN          NaN        NaN   \n",
       "6        NaN           4        NaN         NaN          NaN        NaN   \n",
       "7        NaN           3        NaN         NaN          NaN        NaN   \n",
       "\n",
       "   DF_p20.RD DF_p3.GT  DF_p3.GQ  DF_p3.SDP  DF_p3.DP DF_p3.FREQ    DF_p3.PVAL  \\\n",
       "0        NaN      T/T      87.0         20      20.0        90%  1.675800e-09   \n",
       "2        NaN      C/C      71.0         20      20.0        80%  7.708600e-08   \n",
       "5        NaN      C/C      76.0         18      18.0     88.89%  2.093600e-08   \n",
       "6        NaN      C/C      76.0         14      14.0       100%  2.492700e-08   \n",
       "7        NaN      G/G      64.0         12      12.0       100%  3.698000e-07   \n",
       "\n",
       "   DF_p3.AD  DF_p3.RD DF_p33.GT  DF_p33.GQ  DF_p33.SDP  DF_p33.DP DF_p33.FREQ  \\\n",
       "0      18.0       2.0       ./.        NaN           5        NaN         NaN   \n",
       "2      16.0       4.0       C/C       47.0          11       11.0      90.91%   \n",
       "5      16.0       2.0       T/C       29.0           9        9.0      77.78%   \n",
       "6      14.0       0.0       C/C       58.0          11       11.0        100%   \n",
       "7      12.0       0.0       G/G       46.0           9        9.0        100%   \n",
       "\n",
       "   ...  DF_p86.GT  DF_p86.GQ  DF_p86.SDP DF_p86.DP  DF_p86.FREQ   DF_p86.PVAL  \\\n",
       "0  ...        G/T       29.0           9       9.0       77.78%  1.131200e-03   \n",
       "2  ...        C/C       68.0          18      18.0       83.33%  1.465500e-07   \n",
       "5  ...        C/C       71.0          18      18.0       88.24%  7.327700e-08   \n",
       "6  ...        C/C       71.0          17      17.0       88.24%  7.327700e-08   \n",
       "7  ...        G/G       49.0          13      13.0       84.62%  1.009600e-05   \n",
       "\n",
       "   DF_p86.AD DF_p86.RD  DF_p87.GT  DF_p87.GQ  DF_p87.SDP DF_p87.DP  \\\n",
       "0        7.0       2.0        T/T       44.0          12      12.0   \n",
       "2       15.0       3.0        C/C       81.0          17      17.0   \n",
       "5       15.0       2.0        C/C       52.0          10      10.0   \n",
       "6       15.0       2.0        C/C       52.0          10      10.0   \n",
       "7       11.0       2.0        G/G       41.0           8       8.0   \n",
       "\n",
       "   DF_p87.FREQ   DF_p87.PVAL  DF_p87.AD DF_p87.RD  DF_p88.GT  DF_p88.GQ  \\\n",
       "0       83.33%  3.365200e-05       10.0       2.0        T/T       52.0   \n",
       "2       94.12%  7.713400e-09       16.0       1.0        C/C       60.0   \n",
       "5         100%  5.412500e-06       10.0       0.0        ./.        NaN   \n",
       "6         100%  5.412500e-06       10.0       0.0        C/C       36.0   \n",
       "7         100%  7.770000e-05        8.0       0.0        G/G       46.0   \n",
       "\n",
       "   DF_p88.SDP DF_p88.DP  DF_p88.FREQ   DF_p88.PVAL  DF_p88.AD DF_p88.RD  \\\n",
       "0          10      10.0         100%  5.412500e-06       10.0       0.0   \n",
       "2          15      15.0       86.67%  8.767500e-07       13.0       2.0   \n",
       "5           7       NaN          NaN           NaN        NaN       NaN   \n",
       "6           9       9.0       88.89%  2.056800e-04        8.0       1.0   \n",
       "7           9       9.0         100%  2.056800e-05        9.0       0.0   \n",
       "\n",
       "   DF_p89.GT  DF_p89.GQ  DF_p89.SDP DF_p89.DP  DF_p89.FREQ   DF_p89.PVAL  \\\n",
       "0        T/T       52.0          15      15.0          80%  5.260500e-06   \n",
       "2        T/C       79.0          23      23.0       78.26%  1.193700e-08   \n",
       "5        C/C      105.0          25      25.0          88%  2.591600e-11   \n",
       "6        C/C       86.0          18      18.0       94.44%  2.093600e-09   \n",
       "7        G/G       34.0          10      10.0          80%  3.572300e-04   \n",
       "\n",
       "   DF_p89.AD DF_p89.RD  DF_p9.GT  DF_p9.GQ  DF_p9.SDP DF_p9.DP  DF_p9.FREQ  \\\n",
       "0       12.0       3.0       T/T      39.0         11     11.0      81.82%   \n",
       "2       18.0       5.0       T/C      49.0         17     17.0      70.59%   \n",
       "5       22.0       3.0       T/C      18.0          9      9.0         NaN   \n",
       "6       17.0       1.0       C/C      52.0         10     10.0        100%   \n",
       "7        8.0       2.0       G/G      46.0          9      9.0        100%   \n",
       "\n",
       "   DF_p9.PVAL  DF_p9.AD DF_p9.RD  DF_p94.GT  DF_p94.GQ  DF_p94.SDP DF_p94.DP  \\\n",
       "0    0.000111       9.0      2.0        G/T       31.0          12      12.0   \n",
       "2    0.000011      12.0      5.0        T/C       60.0          18      18.0   \n",
       "5    0.014706       5.0      4.0        C/C       55.0          14      14.0   \n",
       "6    0.000005      10.0      0.0        C/C       52.0          10      10.0   \n",
       "7    0.000021       9.0      0.0        A/G       37.0          12      12.0   \n",
       "\n",
       "   DF_p94.FREQ   DF_p94.PVAL  DF_p94.AD DF_p94.RD                   locus  \\\n",
       "0       66.67%  6.730400e-04        8.0       4.0  jcf7190000000000-77602   \n",
       "2       77.78%  8.060500e-07       14.0       4.0  jcf7190000000000-77657   \n",
       "5       85.71%  2.991300e-06       12.0       2.0  jcf7190000000000-77738   \n",
       "6         100%  5.412500e-06       10.0       0.0  jcf7190000000000-77764   \n",
       "7          75%  1.682600e-04        9.0       3.0  jcf7190000000000-77784   \n",
       "\n",
       "        MAF  \n",
       "0  0.177036  \n",
       "2  0.191005  \n",
       "5  0.195786  \n",
       "6  0.066066  \n",
       "7  0.095813  \n",
       "\n",
       "[5 rows x 351 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# before recalculating SNP data, compare to below\n",
    "interior.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 17%|█▋        | 7/42 [00:00<00:00, 69.41it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2609583, 351)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 42/42 [00:00<00:00, 74.59it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2609583, 351)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AF</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>TYPE</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>ADP</th>\n",
       "      <th>WT</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>NC</th>\n",
       "      <th>DF_p18.GT</th>\n",
       "      <th>DF_p18.GQ</th>\n",
       "      <th>DF_p18.SDP</th>\n",
       "      <th>DF_p18.DP</th>\n",
       "      <th>DF_p18.FREQ</th>\n",
       "      <th>DF_p18.PVAL</th>\n",
       "      <th>DF_p18.AD</th>\n",
       "      <th>DF_p18.RD</th>\n",
       "      <th>DF_p19.GT</th>\n",
       "      <th>DF_p19.GQ</th>\n",
       "      <th>DF_p19.SDP</th>\n",
       "      <th>DF_p19.DP</th>\n",
       "      <th>DF_p19.FREQ</th>\n",
       "      <th>DF_p19.PVAL</th>\n",
       "      <th>DF_p19.AD</th>\n",
       "      <th>DF_p19.RD</th>\n",
       "      <th>DF_p20.GT</th>\n",
       "      <th>DF_p20.GQ</th>\n",
       "      <th>DF_p20.SDP</th>\n",
       "      <th>DF_p20.DP</th>\n",
       "      <th>DF_p20.FREQ</th>\n",
       "      <th>DF_p20.PVAL</th>\n",
       "      <th>DF_p20.AD</th>\n",
       "      <th>DF_p20.RD</th>\n",
       "      <th>DF_p3.GT</th>\n",
       "      <th>DF_p3.GQ</th>\n",
       "      <th>DF_p3.SDP</th>\n",
       "      <th>DF_p3.DP</th>\n",
       "      <th>DF_p3.FREQ</th>\n",
       "      <th>DF_p3.PVAL</th>\n",
       "      <th>DF_p3.AD</th>\n",
       "      <th>DF_p3.RD</th>\n",
       "      <th>DF_p33.GT</th>\n",
       "      <th>DF_p33.GQ</th>\n",
       "      <th>DF_p33.SDP</th>\n",
       "      <th>DF_p33.DP</th>\n",
       "      <th>DF_p33.FREQ</th>\n",
       "      <th>...</th>\n",
       "      <th>DF_p86.GT</th>\n",
       "      <th>DF_p86.GQ</th>\n",
       "      <th>DF_p86.SDP</th>\n",
       "      <th>DF_p86.DP</th>\n",
       "      <th>DF_p86.FREQ</th>\n",
       "      <th>DF_p86.PVAL</th>\n",
       "      <th>DF_p86.AD</th>\n",
       "      <th>DF_p86.RD</th>\n",
       "      <th>DF_p87.GT</th>\n",
       "      <th>DF_p87.GQ</th>\n",
       "      <th>DF_p87.SDP</th>\n",
       "      <th>DF_p87.DP</th>\n",
       "      <th>DF_p87.FREQ</th>\n",
       "      <th>DF_p87.PVAL</th>\n",
       "      <th>DF_p87.AD</th>\n",
       "      <th>DF_p87.RD</th>\n",
       "      <th>DF_p88.GT</th>\n",
       "      <th>DF_p88.GQ</th>\n",
       "      <th>DF_p88.SDP</th>\n",
       "      <th>DF_p88.DP</th>\n",
       "      <th>DF_p88.FREQ</th>\n",
       "      <th>DF_p88.PVAL</th>\n",
       "      <th>DF_p88.AD</th>\n",
       "      <th>DF_p88.RD</th>\n",
       "      <th>DF_p89.GT</th>\n",
       "      <th>DF_p89.GQ</th>\n",
       "      <th>DF_p89.SDP</th>\n",
       "      <th>DF_p89.DP</th>\n",
       "      <th>DF_p89.FREQ</th>\n",
       "      <th>DF_p89.PVAL</th>\n",
       "      <th>DF_p89.AD</th>\n",
       "      <th>DF_p89.RD</th>\n",
       "      <th>DF_p9.GT</th>\n",
       "      <th>DF_p9.GQ</th>\n",
       "      <th>DF_p9.SDP</th>\n",
       "      <th>DF_p9.DP</th>\n",
       "      <th>DF_p9.FREQ</th>\n",
       "      <th>DF_p9.PVAL</th>\n",
       "      <th>DF_p9.AD</th>\n",
       "      <th>DF_p9.RD</th>\n",
       "      <th>DF_p94.GT</th>\n",
       "      <th>DF_p94.GQ</th>\n",
       "      <th>DF_p94.SDP</th>\n",
       "      <th>DF_p94.DP</th>\n",
       "      <th>DF_p94.FREQ</th>\n",
       "      <th>DF_p94.PVAL</th>\n",
       "      <th>DF_p94.AD</th>\n",
       "      <th>DF_p94.RD</th>\n",
       "      <th>locus</th>\n",
       "      <th>MAF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>jcf7190000000000</td>\n",
       "      <td>77602</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.822964</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>SNP</td>\n",
       "      <td>PASS</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td>34</td>\n",
       "      <td>26</td>\n",
       "      <td>21</td>\n",
       "      <td>G/T</td>\n",
       "      <td>84.0</td>\n",
       "      <td>24</td>\n",
       "      <td>24.0</td>\n",
       "      <td>79.17%</td>\n",
       "      <td>3.682600e-09</td>\n",
       "      <td>19.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>T/T</td>\n",
       "      <td>121.0</td>\n",
       "      <td>26</td>\n",
       "      <td>26.0</td>\n",
       "      <td>92.31%</td>\n",
       "      <td>7.622200e-13</td>\n",
       "      <td>24.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T/T</td>\n",
       "      <td>87.0</td>\n",
       "      <td>20</td>\n",
       "      <td>20.0</td>\n",
       "      <td>90%</td>\n",
       "      <td>1.675800e-09</td>\n",
       "      <td>18.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>G/T</td>\n",
       "      <td>29.0</td>\n",
       "      <td>9</td>\n",
       "      <td>9.0</td>\n",
       "      <td>77.78%</td>\n",
       "      <td>1.131200e-03</td>\n",
       "      <td>7.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>T/T</td>\n",
       "      <td>44.0</td>\n",
       "      <td>12</td>\n",
       "      <td>12.0</td>\n",
       "      <td>83.33%</td>\n",
       "      <td>3.365200e-05</td>\n",
       "      <td>10.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>T/T</td>\n",
       "      <td>52.0</td>\n",
       "      <td>10</td>\n",
       "      <td>10.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>5.412500e-06</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>T/T</td>\n",
       "      <td>52.0</td>\n",
       "      <td>15</td>\n",
       "      <td>15.0</td>\n",
       "      <td>80%</td>\n",
       "      <td>5.260500e-06</td>\n",
       "      <td>12.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>T/T</td>\n",
       "      <td>39.0</td>\n",
       "      <td>11</td>\n",
       "      <td>11.0</td>\n",
       "      <td>81.82%</td>\n",
       "      <td>0.000111</td>\n",
       "      <td>9.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>G/T</td>\n",
       "      <td>31.0</td>\n",
       "      <td>12</td>\n",
       "      <td>12.0</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>6.730400e-04</td>\n",
       "      <td>8.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>jcf7190000000000-77602</td>\n",
       "      <td>0.177036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>jcf7190000000000</td>\n",
       "      <td>77657</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.808995</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>SNP</td>\n",
       "      <td>PASS</td>\n",
       "      <td>16</td>\n",
       "      <td>5</td>\n",
       "      <td>51</td>\n",
       "      <td>26</td>\n",
       "      <td>5</td>\n",
       "      <td>C/C</td>\n",
       "      <td>160.0</td>\n",
       "      <td>43</td>\n",
       "      <td>43.0</td>\n",
       "      <td>81.4%</td>\n",
       "      <td>9.593300e-17</td>\n",
       "      <td>35.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>128.0</td>\n",
       "      <td>34</td>\n",
       "      <td>34.0</td>\n",
       "      <td>82.35%</td>\n",
       "      <td>1.349000e-13</td>\n",
       "      <td>28.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C/C</td>\n",
       "      <td>71.0</td>\n",
       "      <td>20</td>\n",
       "      <td>20.0</td>\n",
       "      <td>80%</td>\n",
       "      <td>7.708600e-08</td>\n",
       "      <td>16.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>47.0</td>\n",
       "      <td>11</td>\n",
       "      <td>11.0</td>\n",
       "      <td>90.91%</td>\n",
       "      <td>...</td>\n",
       "      <td>C/C</td>\n",
       "      <td>68.0</td>\n",
       "      <td>18</td>\n",
       "      <td>18.0</td>\n",
       "      <td>83.33%</td>\n",
       "      <td>1.465500e-07</td>\n",
       "      <td>15.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>81.0</td>\n",
       "      <td>17</td>\n",
       "      <td>17.0</td>\n",
       "      <td>94.12%</td>\n",
       "      <td>7.713400e-09</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>60.0</td>\n",
       "      <td>15</td>\n",
       "      <td>15.0</td>\n",
       "      <td>86.67%</td>\n",
       "      <td>8.767500e-07</td>\n",
       "      <td>13.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>79.0</td>\n",
       "      <td>23</td>\n",
       "      <td>23.0</td>\n",
       "      <td>78.26%</td>\n",
       "      <td>1.193700e-08</td>\n",
       "      <td>18.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>49.0</td>\n",
       "      <td>17</td>\n",
       "      <td>17.0</td>\n",
       "      <td>70.59%</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>12.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>60.0</td>\n",
       "      <td>18</td>\n",
       "      <td>18.0</td>\n",
       "      <td>77.78%</td>\n",
       "      <td>8.060500e-07</td>\n",
       "      <td>14.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>jcf7190000000000-77657</td>\n",
       "      <td>0.191005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>jcf7190000000000</td>\n",
       "      <td>77738</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.804214</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>SNP</td>\n",
       "      <td>PASS</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>49</td>\n",
       "      <td>28</td>\n",
       "      <td>8</td>\n",
       "      <td>T/C</td>\n",
       "      <td>103.0</td>\n",
       "      <td>32</td>\n",
       "      <td>32.0</td>\n",
       "      <td>75%</td>\n",
       "      <td>4.196400e-11</td>\n",
       "      <td>24.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>126.0</td>\n",
       "      <td>27</td>\n",
       "      <td>27.0</td>\n",
       "      <td>92.59%</td>\n",
       "      <td>2.085300e-13</td>\n",
       "      <td>25.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C/C</td>\n",
       "      <td>76.0</td>\n",
       "      <td>18</td>\n",
       "      <td>18.0</td>\n",
       "      <td>88.89%</td>\n",
       "      <td>2.093600e-08</td>\n",
       "      <td>16.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>29.0</td>\n",
       "      <td>9</td>\n",
       "      <td>9.0</td>\n",
       "      <td>77.78%</td>\n",
       "      <td>...</td>\n",
       "      <td>C/C</td>\n",
       "      <td>71.0</td>\n",
       "      <td>18</td>\n",
       "      <td>18.0</td>\n",
       "      <td>88.24%</td>\n",
       "      <td>7.327700e-08</td>\n",
       "      <td>15.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>52.0</td>\n",
       "      <td>10</td>\n",
       "      <td>10.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>5.412500e-06</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C/C</td>\n",
       "      <td>105.0</td>\n",
       "      <td>25</td>\n",
       "      <td>25.0</td>\n",
       "      <td>88%</td>\n",
       "      <td>2.591600e-11</td>\n",
       "      <td>22.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>18.0</td>\n",
       "      <td>9</td>\n",
       "      <td>9.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.014706</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>55.0</td>\n",
       "      <td>14</td>\n",
       "      <td>14.0</td>\n",
       "      <td>85.71%</td>\n",
       "      <td>2.991300e-06</td>\n",
       "      <td>12.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>jcf7190000000000-77738</td>\n",
       "      <td>0.195786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>jcf7190000000000</td>\n",
       "      <td>77764</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.933934</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>SNP</td>\n",
       "      <td>PASS</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>28</td>\n",
       "      <td>50</td>\n",
       "      <td>8</td>\n",
       "      <td>C/C</td>\n",
       "      <td>127.0</td>\n",
       "      <td>29</td>\n",
       "      <td>29.0</td>\n",
       "      <td>89.66%</td>\n",
       "      <td>1.649600e-13</td>\n",
       "      <td>26.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>117.0</td>\n",
       "      <td>21</td>\n",
       "      <td>21.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>1.857800e-12</td>\n",
       "      <td>21.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C/C</td>\n",
       "      <td>76.0</td>\n",
       "      <td>14</td>\n",
       "      <td>14.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.492700e-08</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>58.0</td>\n",
       "      <td>11</td>\n",
       "      <td>11.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>...</td>\n",
       "      <td>C/C</td>\n",
       "      <td>71.0</td>\n",
       "      <td>17</td>\n",
       "      <td>17.0</td>\n",
       "      <td>88.24%</td>\n",
       "      <td>7.327700e-08</td>\n",
       "      <td>15.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>52.0</td>\n",
       "      <td>10</td>\n",
       "      <td>10.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>5.412500e-06</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>36.0</td>\n",
       "      <td>9</td>\n",
       "      <td>9.0</td>\n",
       "      <td>88.89%</td>\n",
       "      <td>2.056800e-04</td>\n",
       "      <td>8.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>86.0</td>\n",
       "      <td>18</td>\n",
       "      <td>18.0</td>\n",
       "      <td>94.44%</td>\n",
       "      <td>2.093600e-09</td>\n",
       "      <td>17.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>52.0</td>\n",
       "      <td>10</td>\n",
       "      <td>10.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>52.0</td>\n",
       "      <td>10</td>\n",
       "      <td>10.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>5.412500e-06</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>jcf7190000000000-77764</td>\n",
       "      <td>0.066066</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>jcf7190000000000</td>\n",
       "      <td>77784</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.904187</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>SNP</td>\n",
       "      <td>PASS</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>38</td>\n",
       "      <td>17</td>\n",
       "      <td>G/G</td>\n",
       "      <td>107.0</td>\n",
       "      <td>27</td>\n",
       "      <td>27.0</td>\n",
       "      <td>85.19%</td>\n",
       "      <td>1.616100e-11</td>\n",
       "      <td>23.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>G/G</td>\n",
       "      <td>111.0</td>\n",
       "      <td>20</td>\n",
       "      <td>20.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>7.254400e-12</td>\n",
       "      <td>20.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G/G</td>\n",
       "      <td>64.0</td>\n",
       "      <td>12</td>\n",
       "      <td>12.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>3.698000e-07</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G/G</td>\n",
       "      <td>46.0</td>\n",
       "      <td>9</td>\n",
       "      <td>9.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>...</td>\n",
       "      <td>G/G</td>\n",
       "      <td>49.0</td>\n",
       "      <td>13</td>\n",
       "      <td>13.0</td>\n",
       "      <td>84.62%</td>\n",
       "      <td>1.009600e-05</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>G/G</td>\n",
       "      <td>41.0</td>\n",
       "      <td>8</td>\n",
       "      <td>8.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>7.770000e-05</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G/G</td>\n",
       "      <td>46.0</td>\n",
       "      <td>9</td>\n",
       "      <td>9.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>2.056800e-05</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G/G</td>\n",
       "      <td>34.0</td>\n",
       "      <td>10</td>\n",
       "      <td>10.0</td>\n",
       "      <td>80%</td>\n",
       "      <td>3.572300e-04</td>\n",
       "      <td>8.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>G/G</td>\n",
       "      <td>46.0</td>\n",
       "      <td>9</td>\n",
       "      <td>9.0</td>\n",
       "      <td>100%</td>\n",
       "      <td>0.000021</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>37.0</td>\n",
       "      <td>12</td>\n",
       "      <td>12.0</td>\n",
       "      <td>75%</td>\n",
       "      <td>1.682600e-04</td>\n",
       "      <td>9.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>jcf7190000000000-77784</td>\n",
       "      <td>0.095813</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 351 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              CHROM    POS REF ALT        AF  QUAL TYPE FILTER  ADP  WT  HET  \\\n",
       "0  jcf7190000000000  77602   G   T  0.822964 -10.0  SNP   PASS   11   6   34   \n",
       "2  jcf7190000000000  77657   T   C  0.808995 -10.0  SNP   PASS   16   5   51   \n",
       "5  jcf7190000000000  77738   T   C  0.804214 -10.0  SNP   PASS   13   2   49   \n",
       "6  jcf7190000000000  77764   T   C  0.933934 -10.0  SNP   PASS   12   1   28   \n",
       "7  jcf7190000000000  77784   A   G  0.904187 -10.0  SNP   PASS   11   0   32   \n",
       "\n",
       "   HOM  NC DF_p18.GT  DF_p18.GQ  DF_p18.SDP  DF_p18.DP DF_p18.FREQ  \\\n",
       "0   26  21       G/T       84.0          24       24.0      79.17%   \n",
       "2   26   5       C/C      160.0          43       43.0       81.4%   \n",
       "5   28   8       T/C      103.0          32       32.0         75%   \n",
       "6   50   8       C/C      127.0          29       29.0      89.66%   \n",
       "7   38  17       G/G      107.0          27       27.0      85.19%   \n",
       "\n",
       "    DF_p18.PVAL  DF_p18.AD  DF_p18.RD DF_p19.GT  DF_p19.GQ  DF_p19.SDP  \\\n",
       "0  3.682600e-09       19.0        5.0       T/T      121.0          26   \n",
       "2  9.593300e-17       35.0        8.0       C/C      128.0          34   \n",
       "5  4.196400e-11       24.0        8.0       C/C      126.0          27   \n",
       "6  1.649600e-13       26.0        3.0       C/C      117.0          21   \n",
       "7  1.616100e-11       23.0        4.0       G/G      111.0          20   \n",
       "\n",
       "   DF_p19.DP DF_p19.FREQ   DF_p19.PVAL  DF_p19.AD  DF_p19.RD DF_p20.GT  \\\n",
       "0       26.0      92.31%  7.622200e-13       24.0        2.0       ./.   \n",
       "2       34.0      82.35%  1.349000e-13       28.0        6.0       ./.   \n",
       "5       27.0      92.59%  2.085300e-13       25.0        2.0       ./.   \n",
       "6       21.0        100%  1.857800e-12       21.0        0.0       ./.   \n",
       "7       20.0        100%  7.254400e-12       20.0        0.0       ./.   \n",
       "\n",
       "   DF_p20.GQ  DF_p20.SDP  DF_p20.DP DF_p20.FREQ  DF_p20.PVAL  DF_p20.AD  \\\n",
       "0        NaN           2        NaN         NaN          NaN        NaN   \n",
       "2        NaN           3        NaN         NaN          NaN        NaN   \n",
       "5        NaN           3        NaN         NaN          NaN        NaN   \n",
       "6        NaN           4        NaN         NaN          NaN        NaN   \n",
       "7        NaN           3        NaN         NaN          NaN        NaN   \n",
       "\n",
       "   DF_p20.RD DF_p3.GT  DF_p3.GQ  DF_p3.SDP  DF_p3.DP DF_p3.FREQ    DF_p3.PVAL  \\\n",
       "0        NaN      T/T      87.0         20      20.0        90%  1.675800e-09   \n",
       "2        NaN      C/C      71.0         20      20.0        80%  7.708600e-08   \n",
       "5        NaN      C/C      76.0         18      18.0     88.89%  2.093600e-08   \n",
       "6        NaN      C/C      76.0         14      14.0       100%  2.492700e-08   \n",
       "7        NaN      G/G      64.0         12      12.0       100%  3.698000e-07   \n",
       "\n",
       "   DF_p3.AD  DF_p3.RD DF_p33.GT  DF_p33.GQ  DF_p33.SDP  DF_p33.DP DF_p33.FREQ  \\\n",
       "0      18.0       2.0       ./.        NaN           5        NaN         NaN   \n",
       "2      16.0       4.0       C/C       47.0          11       11.0      90.91%   \n",
       "5      16.0       2.0       T/C       29.0           9        9.0      77.78%   \n",
       "6      14.0       0.0       C/C       58.0          11       11.0        100%   \n",
       "7      12.0       0.0       G/G       46.0           9        9.0        100%   \n",
       "\n",
       "   ...  DF_p86.GT  DF_p86.GQ  DF_p86.SDP DF_p86.DP  DF_p86.FREQ   DF_p86.PVAL  \\\n",
       "0  ...        G/T       29.0           9       9.0       77.78%  1.131200e-03   \n",
       "2  ...        C/C       68.0          18      18.0       83.33%  1.465500e-07   \n",
       "5  ...        C/C       71.0          18      18.0       88.24%  7.327700e-08   \n",
       "6  ...        C/C       71.0          17      17.0       88.24%  7.327700e-08   \n",
       "7  ...        G/G       49.0          13      13.0       84.62%  1.009600e-05   \n",
       "\n",
       "   DF_p86.AD DF_p86.RD  DF_p87.GT  DF_p87.GQ  DF_p87.SDP DF_p87.DP  \\\n",
       "0        7.0       2.0        T/T       44.0          12      12.0   \n",
       "2       15.0       3.0        C/C       81.0          17      17.0   \n",
       "5       15.0       3.0        C/C       52.0          10      10.0   \n",
       "6       15.0       2.0        C/C       52.0          10      10.0   \n",
       "7       11.0       2.0        G/G       41.0           8       8.0   \n",
       "\n",
       "   DF_p87.FREQ   DF_p87.PVAL  DF_p87.AD DF_p87.RD  DF_p88.GT  DF_p88.GQ  \\\n",
       "0       83.33%  3.365200e-05       10.0       2.0        T/T       52.0   \n",
       "2       94.12%  7.713400e-09       16.0       1.0        C/C       60.0   \n",
       "5         100%  5.412500e-06       10.0       0.0        ./.        NaN   \n",
       "6         100%  5.412500e-06       10.0       0.0        C/C       36.0   \n",
       "7         100%  7.770000e-05        8.0       0.0        G/G       46.0   \n",
       "\n",
       "   DF_p88.SDP DF_p88.DP  DF_p88.FREQ   DF_p88.PVAL  DF_p88.AD DF_p88.RD  \\\n",
       "0          10      10.0         100%  5.412500e-06       10.0       0.0   \n",
       "2          15      15.0       86.67%  8.767500e-07       13.0       2.0   \n",
       "5           7       NaN          NaN           NaN        NaN       NaN   \n",
       "6           9       9.0       88.89%  2.056800e-04        8.0       1.0   \n",
       "7           9       9.0         100%  2.056800e-05        9.0       0.0   \n",
       "\n",
       "   DF_p89.GT  DF_p89.GQ  DF_p89.SDP DF_p89.DP  DF_p89.FREQ   DF_p89.PVAL  \\\n",
       "0        T/T       52.0          15      15.0          80%  5.260500e-06   \n",
       "2        T/C       79.0          23      23.0       78.26%  1.193700e-08   \n",
       "5        C/C      105.0          25      25.0          88%  2.591600e-11   \n",
       "6        C/C       86.0          18      18.0       94.44%  2.093600e-09   \n",
       "7        G/G       34.0          10      10.0          80%  3.572300e-04   \n",
       "\n",
       "   DF_p89.AD DF_p89.RD  DF_p9.GT  DF_p9.GQ  DF_p9.SDP DF_p9.DP  DF_p9.FREQ  \\\n",
       "0       12.0       3.0       T/T      39.0         11     11.0      81.82%   \n",
       "2       18.0       5.0       T/C      49.0         17     17.0      70.59%   \n",
       "5       22.0       3.0       T/C      18.0          9      9.0         NaN   \n",
       "6       17.0       1.0       C/C      52.0         10     10.0        100%   \n",
       "7        8.0       2.0       G/G      46.0          9      9.0        100%   \n",
       "\n",
       "   DF_p9.PVAL  DF_p9.AD DF_p9.RD  DF_p94.GT  DF_p94.GQ  DF_p94.SDP DF_p94.DP  \\\n",
       "0    0.000111       9.0      2.0        G/T       31.0          12      12.0   \n",
       "2    0.000011      12.0      5.0        T/C       60.0          18      18.0   \n",
       "5    0.014706       5.0      4.0        C/C       55.0          14      14.0   \n",
       "6    0.000005      10.0      0.0        C/C       52.0          10      10.0   \n",
       "7    0.000021       9.0      0.0        A/G       37.0          12      12.0   \n",
       "\n",
       "   DF_p94.FREQ   DF_p94.PVAL  DF_p94.AD DF_p94.RD                   locus  \\\n",
       "0       66.67%  6.730400e-04        8.0       4.0  jcf7190000000000-77602   \n",
       "2       77.78%  8.060500e-07       14.0       4.0  jcf7190000000000-77657   \n",
       "5       85.71%  2.991300e-06       12.0       2.0  jcf7190000000000-77738   \n",
       "6         100%  5.412500e-06       10.0       0.0  jcf7190000000000-77764   \n",
       "7          75%  1.682600e-04        9.0       3.0  jcf7190000000000-77784   \n",
       "\n",
       "        MAF  \n",
       "0  0.177036  \n",
       "2  0.191005  \n",
       "5  0.195786  \n",
       "6  0.066066  \n",
       "7  0.095813  \n",
       "\n",
       "[5 rows x 351 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# recalculate SNP data\n",
    "print(interior.shape)\n",
    "interior_recalc = recalc_rd(interior)\n",
    "print(interior_recalc.shape)\n",
    "interior_recalc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "# save interior SNP data\n",
    "intfile = op.join(newdirs['FDI'], 'DF_pooled-varscan_all_bedfiles_SNP_FDI_maf_RD-recalculated.txt')\n",
    "write_jobs = [lview.apply_async(save_file, *(interior_recalc, intfile))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 42/42 [00:00<00:00, 3787.51it/s]\n",
      "100%|██████████| 42/42 [00:00<00:00, 1378.00it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DF_pooled-varscan_all_bedfiles_PARALOGS_FDI.txt (988, 351)\n",
      "\t (988, 351)\n",
      "DF_pooled-varscan_all_bedfiles_REPEATS_FDI.txt (89609, 351)\n",
      "\t (89609, 351)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "# recalc RD for paralogs and repeats\n",
    "fdi_recalc = {}\n",
    "for key,df in fdi_remaining.items():\n",
    "    print(op.basename(key), df.shape)\n",
    "    fdi_recalc[key] = recalc_rd(df)\n",
    "    print('\\t', fdi_recalc[key].shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'FDI': '/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/03_maf-p05_RD-recalculated_FDI',\n",
       " 'FDC': '/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/03_maf-p05_RD-recalculated_FDC'}"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "newdirs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DF_pooled-varscan_all_bedfiles_PARALOGS_FDI_maf_RD-recalculated.txt\n",
      "DF_pooled-varscan_all_bedfiles_REPEATS_FDI_maf_RD-recalculated.txt\n"
     ]
    }
   ],
   "source": [
    "# save paralog and repeats\n",
    "for key,df in fdi_recalc.items():\n",
    "    dst = op.join(newdirs['FDI'], op.basename(key).replace(\".txt\", \"_maf_RD-recalculated.txt\"))\n",
    "    print(op.basename(dst))\n",
    "    save_file(df, dst)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### now recalc coastal files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AF</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>TYPE</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>ADP</th>\n",
       "      <th>WT</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>NC</th>\n",
       "      <th>DF_p1.GT</th>\n",
       "      <th>DF_p1.GQ</th>\n",
       "      <th>DF_p1.SDP</th>\n",
       "      <th>DF_p1.DP</th>\n",
       "      <th>DF_p1.FREQ</th>\n",
       "      <th>DF_p1.PVAL</th>\n",
       "      <th>DF_p1.AD</th>\n",
       "      <th>DF_p1.RD</th>\n",
       "      <th>DF_p2.GT</th>\n",
       "      <th>DF_p2.GQ</th>\n",
       "      <th>DF_p2.SDP</th>\n",
       "      <th>DF_p2.DP</th>\n",
       "      <th>DF_p2.FREQ</th>\n",
       "      <th>DF_p2.PVAL</th>\n",
       "      <th>DF_p2.AD</th>\n",
       "      <th>DF_p2.RD</th>\n",
       "      <th>DF_p23.GT</th>\n",
       "      <th>DF_p23.GQ</th>\n",
       "      <th>DF_p23.SDP</th>\n",
       "      <th>DF_p23.DP</th>\n",
       "      <th>DF_p23.FREQ</th>\n",
       "      <th>DF_p23.PVAL</th>\n",
       "      <th>DF_p23.AD</th>\n",
       "      <th>DF_p23.RD</th>\n",
       "      <th>DF_p24.GT</th>\n",
       "      <th>DF_p24.GQ</th>\n",
       "      <th>DF_p24.SDP</th>\n",
       "      <th>DF_p24.DP</th>\n",
       "      <th>DF_p24.FREQ</th>\n",
       "      <th>DF_p24.PVAL</th>\n",
       "      <th>DF_p24.AD</th>\n",
       "      <th>DF_p24.RD</th>\n",
       "      <th>DF_p25.GT</th>\n",
       "      <th>DF_p25.GQ</th>\n",
       "      <th>DF_p25.SDP</th>\n",
       "      <th>DF_p25.DP</th>\n",
       "      <th>DF_p25.FREQ</th>\n",
       "      <th>...</th>\n",
       "      <th>DF_p93.GT</th>\n",
       "      <th>DF_p93.GQ</th>\n",
       "      <th>DF_p93.SDP</th>\n",
       "      <th>DF_p93.DP</th>\n",
       "      <th>DF_p93.FREQ</th>\n",
       "      <th>DF_p93.PVAL</th>\n",
       "      <th>DF_p93.AD</th>\n",
       "      <th>DF_p93.RD</th>\n",
       "      <th>DF_p95.GT</th>\n",
       "      <th>DF_p95.GQ</th>\n",
       "      <th>DF_p95.SDP</th>\n",
       "      <th>DF_p95.DP</th>\n",
       "      <th>DF_p95.FREQ</th>\n",
       "      <th>DF_p95.PVAL</th>\n",
       "      <th>DF_p95.AD</th>\n",
       "      <th>DF_p95.RD</th>\n",
       "      <th>DF_p96.GT</th>\n",
       "      <th>DF_p96.GQ</th>\n",
       "      <th>DF_p96.SDP</th>\n",
       "      <th>DF_p96.DP</th>\n",
       "      <th>DF_p96.FREQ</th>\n",
       "      <th>DF_p96.PVAL</th>\n",
       "      <th>DF_p96.AD</th>\n",
       "      <th>DF_p96.RD</th>\n",
       "      <th>DF_p97.GT</th>\n",
       "      <th>DF_p97.GQ</th>\n",
       "      <th>DF_p97.SDP</th>\n",
       "      <th>DF_p97.DP</th>\n",
       "      <th>DF_p97.FREQ</th>\n",
       "      <th>DF_p97.PVAL</th>\n",
       "      <th>DF_p97.AD</th>\n",
       "      <th>DF_p97.RD</th>\n",
       "      <th>DF_p98.GT</th>\n",
       "      <th>DF_p98.GQ</th>\n",
       "      <th>DF_p98.SDP</th>\n",
       "      <th>DF_p98.DP</th>\n",
       "      <th>DF_p98.FREQ</th>\n",
       "      <th>DF_p98.PVAL</th>\n",
       "      <th>DF_p98.AD</th>\n",
       "      <th>DF_p98.RD</th>\n",
       "      <th>DF_p17.GT</th>\n",
       "      <th>DF_p17.GQ</th>\n",
       "      <th>DF_p17.SDP</th>\n",
       "      <th>DF_p17.DP</th>\n",
       "      <th>DF_p17.FREQ</th>\n",
       "      <th>DF_p17.PVAL</th>\n",
       "      <th>DF_p17.AD</th>\n",
       "      <th>DF_p17.RD</th>\n",
       "      <th>locus</th>\n",
       "      <th>MAF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>jcf7190000000000</td>\n",
       "      <td>77738</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.668265</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>SNP</td>\n",
       "      <td>PASS</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>49</td>\n",
       "      <td>28</td>\n",
       "      <td>8</td>\n",
       "      <td>T/C</td>\n",
       "      <td>49.0</td>\n",
       "      <td>17</td>\n",
       "      <td>17.0</td>\n",
       "      <td>70.59%</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>12.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T/C</td>\n",
       "      <td>18.0</td>\n",
       "      <td>8</td>\n",
       "      <td>8.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.012821</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>32.0</td>\n",
       "      <td>11</td>\n",
       "      <td>11.0</td>\n",
       "      <td>72.73%</td>\n",
       "      <td>5.160000e-04</td>\n",
       "      <td>8.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>29.0</td>\n",
       "      <td>15</td>\n",
       "      <td>15.0</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>...</td>\n",
       "      <td>T/C</td>\n",
       "      <td>46.0</td>\n",
       "      <td>20</td>\n",
       "      <td>20.0</td>\n",
       "      <td>60%</td>\n",
       "      <td>2.254800e-05</td>\n",
       "      <td>12.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>40.0</td>\n",
       "      <td>15</td>\n",
       "      <td>15.0</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>9.995000e-05</td>\n",
       "      <td>10.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>50.0</td>\n",
       "      <td>22</td>\n",
       "      <td>22.0</td>\n",
       "      <td>59.09%</td>\n",
       "      <td>9.581300e-06</td>\n",
       "      <td>13.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>60.0</td>\n",
       "      <td>18</td>\n",
       "      <td>18.0</td>\n",
       "      <td>77.78%</td>\n",
       "      <td>8.060500e-07</td>\n",
       "      <td>14.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T/C</td>\n",
       "      <td>32.0</td>\n",
       "      <td>11</td>\n",
       "      <td>11.0</td>\n",
       "      <td>72.73%</td>\n",
       "      <td>0.000516</td>\n",
       "      <td>8.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>jcf7190000000000-77738</td>\n",
       "      <td>0.331735</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>jcf7190000000000</td>\n",
       "      <td>77764</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.753734</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>SNP</td>\n",
       "      <td>PASS</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>28</td>\n",
       "      <td>50</td>\n",
       "      <td>8</td>\n",
       "      <td>T/C</td>\n",
       "      <td>50.0</td>\n",
       "      <td>16</td>\n",
       "      <td>16.0</td>\n",
       "      <td>75%</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>12.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C/C</td>\n",
       "      <td>49.0</td>\n",
       "      <td>13</td>\n",
       "      <td>13.0</td>\n",
       "      <td>84.62%</td>\n",
       "      <td>1.009600e-05</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>34.0</td>\n",
       "      <td>15</td>\n",
       "      <td>15.0</td>\n",
       "      <td>60%</td>\n",
       "      <td>...</td>\n",
       "      <td>T/C</td>\n",
       "      <td>53.0</td>\n",
       "      <td>19</td>\n",
       "      <td>19.0</td>\n",
       "      <td>68.42%</td>\n",
       "      <td>5.010600e-06</td>\n",
       "      <td>13.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>46.0</td>\n",
       "      <td>15</td>\n",
       "      <td>15.0</td>\n",
       "      <td>73.33%</td>\n",
       "      <td>2.498800e-05</td>\n",
       "      <td>11.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>75.0</td>\n",
       "      <td>25</td>\n",
       "      <td>25.0</td>\n",
       "      <td>72%</td>\n",
       "      <td>2.662600e-08</td>\n",
       "      <td>18.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>65.0</td>\n",
       "      <td>16</td>\n",
       "      <td>16.0</td>\n",
       "      <td>87.5%</td>\n",
       "      <td>2.545400e-07</td>\n",
       "      <td>14.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C/C</td>\n",
       "      <td>44.0</td>\n",
       "      <td>12</td>\n",
       "      <td>12.0</td>\n",
       "      <td>83.33%</td>\n",
       "      <td>0.000034</td>\n",
       "      <td>10.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>jcf7190000000000-77764</td>\n",
       "      <td>0.246266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>jcf7190000000004</td>\n",
       "      <td>54418</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.367180</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>SNP</td>\n",
       "      <td>PASS</td>\n",
       "      <td>36</td>\n",
       "      <td>1</td>\n",
       "      <td>86</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>39.0</td>\n",
       "      <td>47</td>\n",
       "      <td>47.0</td>\n",
       "      <td>25.53%</td>\n",
       "      <td>0.000109</td>\n",
       "      <td>12.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>27.0</td>\n",
       "      <td>20</td>\n",
       "      <td>20.0</td>\n",
       "      <td>40%</td>\n",
       "      <td>0.001638</td>\n",
       "      <td>8.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>26.0</td>\n",
       "      <td>31</td>\n",
       "      <td>31.0</td>\n",
       "      <td>25.81%</td>\n",
       "      <td>0.002333</td>\n",
       "      <td>8.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>32.0</td>\n",
       "      <td>43</td>\n",
       "      <td>43.0</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.414800e-04</td>\n",
       "      <td>10.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>47.0</td>\n",
       "      <td>42</td>\n",
       "      <td>42.0</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>...</td>\n",
       "      <td>T/C</td>\n",
       "      <td>72.0</td>\n",
       "      <td>45</td>\n",
       "      <td>45.0</td>\n",
       "      <td>44.44%</td>\n",
       "      <td>6.217800e-08</td>\n",
       "      <td>20.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>83.0</td>\n",
       "      <td>50</td>\n",
       "      <td>50.0</td>\n",
       "      <td>46%</td>\n",
       "      <td>4.345100e-09</td>\n",
       "      <td>23.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>29.0</td>\n",
       "      <td>36</td>\n",
       "      <td>36.0</td>\n",
       "      <td>25%</td>\n",
       "      <td>1.106100e-03</td>\n",
       "      <td>9.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>68.0</td>\n",
       "      <td>43</td>\n",
       "      <td>43.0</td>\n",
       "      <td>45.24%</td>\n",
       "      <td>1.352500e-07</td>\n",
       "      <td>19.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>76.0</td>\n",
       "      <td>45</td>\n",
       "      <td>45.0</td>\n",
       "      <td>46.67%</td>\n",
       "      <td>2.220600e-08</td>\n",
       "      <td>21.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>T/T</td>\n",
       "      <td>9.0</td>\n",
       "      <td>14</td>\n",
       "      <td>14.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.111110</td>\n",
       "      <td>3.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>jcf7190000000004-54418</td>\n",
       "      <td>0.367180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>jcf7190000000004</td>\n",
       "      <td>54508</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.080056</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>SNP</td>\n",
       "      <td>PASS</td>\n",
       "      <td>53</td>\n",
       "      <td>67</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>A/A</td>\n",
       "      <td>89.0</td>\n",
       "      <td>65</td>\n",
       "      <td>65.0</td>\n",
       "      <td>4.62%</td>\n",
       "      <td>0.122090</td>\n",
       "      <td>3.0</td>\n",
       "      <td>62.0</td>\n",
       "      <td>A/A</td>\n",
       "      <td>40.0</td>\n",
       "      <td>32</td>\n",
       "      <td>32.0</td>\n",
       "      <td>6.25%</td>\n",
       "      <td>0.246030</td>\n",
       "      <td>2.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>22.0</td>\n",
       "      <td>44</td>\n",
       "      <td>44.0</td>\n",
       "      <td>15.91%</td>\n",
       "      <td>0.006036</td>\n",
       "      <td>7.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>21.0</td>\n",
       "      <td>65</td>\n",
       "      <td>65.0</td>\n",
       "      <td>10.77%</td>\n",
       "      <td>6.590400e-03</td>\n",
       "      <td>7.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>15.0</td>\n",
       "      <td>67</td>\n",
       "      <td>67.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>A/A</td>\n",
       "      <td>85.0</td>\n",
       "      <td>67</td>\n",
       "      <td>67.0</td>\n",
       "      <td>5.97%</td>\n",
       "      <td>5.969100e-02</td>\n",
       "      <td>4.0</td>\n",
       "      <td>63.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>15.0</td>\n",
       "      <td>82</td>\n",
       "      <td>82.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.935100e-02</td>\n",
       "      <td>5.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>A/A</td>\n",
       "      <td>126.0</td>\n",
       "      <td>68</td>\n",
       "      <td>68.0</td>\n",
       "      <td>0%</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>22.0</td>\n",
       "      <td>52</td>\n",
       "      <td>52.0</td>\n",
       "      <td>13.46%</td>\n",
       "      <td>6.297700e-03</td>\n",
       "      <td>7.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>35.0</td>\n",
       "      <td>59</td>\n",
       "      <td>59.0</td>\n",
       "      <td>18.64%</td>\n",
       "      <td>2.925000e-04</td>\n",
       "      <td>11.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>A/A</td>\n",
       "      <td>28.0</td>\n",
       "      <td>28</td>\n",
       "      <td>28.0</td>\n",
       "      <td>10.71%</td>\n",
       "      <td>0.118180</td>\n",
       "      <td>3.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>jcf7190000000004-54508</td>\n",
       "      <td>0.080056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>jcf7190000000004</td>\n",
       "      <td>54540</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.333741</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>SNP</td>\n",
       "      <td>PASS</td>\n",
       "      <td>56</td>\n",
       "      <td>1</td>\n",
       "      <td>58</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>56.0</td>\n",
       "      <td>67</td>\n",
       "      <td>67.0</td>\n",
       "      <td>25.37%</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>17.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>37.0</td>\n",
       "      <td>36</td>\n",
       "      <td>36.0</td>\n",
       "      <td>30.56%</td>\n",
       "      <td>0.000199</td>\n",
       "      <td>11.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>18.0</td>\n",
       "      <td>47</td>\n",
       "      <td>47.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.013188</td>\n",
       "      <td>6.0</td>\n",
       "      <td>41.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>83.0</td>\n",
       "      <td>67</td>\n",
       "      <td>67.0</td>\n",
       "      <td>35.82%</td>\n",
       "      <td>4.810900e-09</td>\n",
       "      <td>24.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>86.0</td>\n",
       "      <td>70</td>\n",
       "      <td>70.0</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>...</td>\n",
       "      <td>A/G</td>\n",
       "      <td>67.0</td>\n",
       "      <td>67</td>\n",
       "      <td>67.0</td>\n",
       "      <td>29.85%</td>\n",
       "      <td>1.799800e-07</td>\n",
       "      <td>20.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>58.0</td>\n",
       "      <td>82</td>\n",
       "      <td>82.0</td>\n",
       "      <td>21.95%</td>\n",
       "      <td>1.339100e-06</td>\n",
       "      <td>18.0</td>\n",
       "      <td>64.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>45.0</td>\n",
       "      <td>69</td>\n",
       "      <td>69.0</td>\n",
       "      <td>20.29%</td>\n",
       "      <td>2.934400e-05</td>\n",
       "      <td>14.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>53.0</td>\n",
       "      <td>58</td>\n",
       "      <td>58.0</td>\n",
       "      <td>27.59%</td>\n",
       "      <td>4.601500e-06</td>\n",
       "      <td>16.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>88.0</td>\n",
       "      <td>62</td>\n",
       "      <td>62.0</td>\n",
       "      <td>40.32%</td>\n",
       "      <td>1.416800e-09</td>\n",
       "      <td>25.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>15.0</td>\n",
       "      <td>26</td>\n",
       "      <td>26.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.025310</td>\n",
       "      <td>5.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>jcf7190000000004-54540</td>\n",
       "      <td>0.333741</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 375 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              CHROM    POS REF ALT        AF  QUAL TYPE FILTER  ADP  WT  HET  \\\n",
       "0  jcf7190000000000  77738   T   C  0.668265 -10.0  SNP   PASS   13   2   49   \n",
       "1  jcf7190000000000  77764   T   C  0.753734 -10.0  SNP   PASS   12   1   28   \n",
       "3  jcf7190000000004  54418   T   C  0.367180 -10.0  SNP   PASS   36   1   86   \n",
       "5  jcf7190000000004  54508   A   G  0.080056 -10.0  SNP   PASS   53  67   20   \n",
       "6  jcf7190000000004  54540   A   G  0.333741 -10.0  SNP   PASS   56   1   58   \n",
       "\n",
       "   HOM  NC DF_p1.GT  DF_p1.GQ  DF_p1.SDP  DF_p1.DP DF_p1.FREQ  DF_p1.PVAL  \\\n",
       "0   28   8      T/C      49.0         17      17.0     70.59%    0.000011   \n",
       "1   50   8      T/C      50.0         16      16.0        75%    0.000008   \n",
       "3    0   0      T/C      39.0         47      47.0     25.53%    0.000109   \n",
       "5    0   0      A/A      89.0         65      65.0      4.62%    0.122090   \n",
       "6   28   0      A/G      56.0         67      67.0     25.37%    0.000002   \n",
       "\n",
       "   DF_p1.AD  DF_p1.RD DF_p2.GT  DF_p2.GQ  DF_p2.SDP  DF_p2.DP DF_p2.FREQ  \\\n",
       "0      12.0       5.0      ./.       NaN          6       NaN        NaN   \n",
       "1      12.0       4.0      ./.       NaN          4       NaN        NaN   \n",
       "3      12.0      35.0      T/C      27.0         20      20.0        40%   \n",
       "5       3.0      62.0      A/A      40.0         32      32.0      6.25%   \n",
       "6      17.0      50.0      A/G      37.0         36      36.0     30.56%   \n",
       "\n",
       "   DF_p2.PVAL  DF_p2.AD  DF_p2.RD DF_p23.GT  DF_p23.GQ  DF_p23.SDP  DF_p23.DP  \\\n",
       "0         NaN       NaN       NaN       T/C       18.0           8        8.0   \n",
       "1         NaN       NaN       NaN       ./.        NaN           6        NaN   \n",
       "3    0.001638       8.0      12.0       T/C       26.0          31       31.0   \n",
       "5    0.246030       2.0      30.0       A/G       22.0          44       44.0   \n",
       "6    0.000199      11.0      25.0       A/G       18.0          47       47.0   \n",
       "\n",
       "  DF_p23.FREQ  DF_p23.PVAL  DF_p23.AD  DF_p23.RD DF_p24.GT  DF_p24.GQ  \\\n",
       "0         NaN     0.012821        5.0        3.0       T/C       32.0   \n",
       "1         NaN          NaN        NaN        NaN       C/C       49.0   \n",
       "3      25.81%     0.002333        8.0       23.0       T/C       32.0   \n",
       "5      15.91%     0.006036        7.0       37.0       A/G       21.0   \n",
       "6         NaN     0.013188        6.0       41.0       A/G       83.0   \n",
       "\n",
       "   DF_p24.SDP  DF_p24.DP DF_p24.FREQ   DF_p24.PVAL  DF_p24.AD  DF_p24.RD  \\\n",
       "0          11       11.0      72.73%  5.160000e-04        8.0        3.0   \n",
       "1          13       13.0      84.62%  1.009600e-05       11.0        2.0   \n",
       "3          43       43.0      23.26%  5.414800e-04       10.0       33.0   \n",
       "5          65       65.0      10.77%  6.590400e-03        7.0       58.0   \n",
       "6          67       67.0      35.82%  4.810900e-09       24.0       43.0   \n",
       "\n",
       "  DF_p25.GT  DF_p25.GQ  DF_p25.SDP  DF_p25.DP DF_p25.FREQ  ...  DF_p93.GT  \\\n",
       "0       T/C       29.0          15       15.0      53.33%  ...        T/C   \n",
       "1       T/C       34.0          15       15.0         60%  ...        T/C   \n",
       "3       T/C       47.0          42       42.0      33.33%  ...        T/C   \n",
       "5       A/G       15.0          67       67.0         NaN  ...        A/A   \n",
       "6       A/G       86.0          70       70.0      35.71%  ...        A/G   \n",
       "\n",
       "   DF_p93.GQ  DF_p93.SDP DF_p93.DP  DF_p93.FREQ   DF_p93.PVAL  DF_p93.AD  \\\n",
       "0       46.0          20      20.0          60%  2.254800e-05       12.0   \n",
       "1       53.0          19      19.0       68.42%  5.010600e-06       13.0   \n",
       "3       72.0          45      45.0       44.44%  6.217800e-08       20.0   \n",
       "5       85.0          67      67.0        5.97%  5.969100e-02        4.0   \n",
       "6       67.0          67      67.0       29.85%  1.799800e-07       20.0   \n",
       "\n",
       "  DF_p93.RD  DF_p95.GT  DF_p95.GQ  DF_p95.SDP DF_p95.DP  DF_p95.FREQ  \\\n",
       "0       8.0        T/C       40.0          15      15.0       66.67%   \n",
       "1       6.0        T/C       46.0          15      15.0       73.33%   \n",
       "3      25.0        T/C       83.0          50      50.0          46%   \n",
       "5      63.0        A/G       15.0          82      82.0          NaN   \n",
       "6      47.0        A/G       58.0          82      82.0       21.95%   \n",
       "\n",
       "    DF_p95.PVAL  DF_p95.AD DF_p95.RD  DF_p96.GT  DF_p96.GQ  DF_p96.SDP  \\\n",
       "0  9.995000e-05       10.0       5.0        T/C       50.0          22   \n",
       "1  2.498800e-05       11.0       4.0        T/C       75.0          25   \n",
       "3  4.345100e-09       23.0      27.0        T/C       29.0          36   \n",
       "5  2.935100e-02        5.0      77.0        A/A      126.0          68   \n",
       "6  1.339100e-06       18.0      64.0        A/G       45.0          69   \n",
       "\n",
       "  DF_p96.DP  DF_p96.FREQ   DF_p96.PVAL  DF_p96.AD DF_p96.RD  DF_p97.GT  \\\n",
       "0      22.0       59.09%  9.581300e-06       13.0       9.0        T/C   \n",
       "1      25.0          72%  2.662600e-08       18.0       7.0        C/C   \n",
       "3      36.0          25%  1.106100e-03        9.0      27.0        T/C   \n",
       "5      68.0           0%  1.000000e+00        0.0      68.0        A/G   \n",
       "6      69.0       20.29%  2.934400e-05       14.0      55.0        A/G   \n",
       "\n",
       "   DF_p97.GQ  DF_p97.SDP DF_p97.DP  DF_p97.FREQ   DF_p97.PVAL  DF_p97.AD  \\\n",
       "0       60.0          18      18.0       77.78%  8.060500e-07       14.0   \n",
       "1       65.0          16      16.0        87.5%  2.545400e-07       14.0   \n",
       "3       68.0          43      43.0       45.24%  1.352500e-07       19.0   \n",
       "5       22.0          52      52.0       13.46%  6.297700e-03        7.0   \n",
       "6       53.0          58      58.0       27.59%  4.601500e-06       16.0   \n",
       "\n",
       "  DF_p97.RD  DF_p98.GT  DF_p98.GQ  DF_p98.SDP DF_p98.DP  DF_p98.FREQ  \\\n",
       "0       4.0        ./.        NaN           6       NaN          NaN   \n",
       "1       2.0        ./.        NaN           4       NaN          NaN   \n",
       "3      23.0        T/C       76.0          45      45.0       46.67%   \n",
       "5      45.0        A/G       35.0          59      59.0       18.64%   \n",
       "6      42.0        A/G       88.0          62      62.0       40.32%   \n",
       "\n",
       "    DF_p98.PVAL  DF_p98.AD DF_p98.RD  DF_p17.GT  DF_p17.GQ  DF_p17.SDP  \\\n",
       "0           NaN        NaN       NaN        T/C       32.0          11   \n",
       "1           NaN        NaN       NaN        C/C       44.0          12   \n",
       "3  2.220600e-08       21.0      24.0        T/T        9.0          14   \n",
       "5  2.925000e-04       11.0      48.0        A/A       28.0          28   \n",
       "6  1.416800e-09       25.0      37.0        A/G       15.0          26   \n",
       "\n",
       "  DF_p17.DP  DF_p17.FREQ  DF_p17.PVAL  DF_p17.AD DF_p17.RD  \\\n",
       "0      11.0       72.73%     0.000516        8.0       3.0   \n",
       "1      12.0       83.33%     0.000034       10.0       2.0   \n",
       "3      14.0          NaN     0.111110        3.0      11.0   \n",
       "5      28.0       10.71%     0.118180        3.0      25.0   \n",
       "6      26.0          NaN     0.025310        5.0      21.0   \n",
       "\n",
       "                    locus       MAF  \n",
       "0  jcf7190000000000-77738  0.331735  \n",
       "1  jcf7190000000000-77764  0.246266  \n",
       "3  jcf7190000000004-54418  0.367180  \n",
       "5  jcf7190000000004-54508  0.080056  \n",
       "6  jcf7190000000004-54540  0.333741  \n",
       "\n",
       "[5 rows x 375 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# before recalculating, compare to below\n",
    "coastal.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 20%|██        | 9/45 [00:00<00:00, 82.09it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2350673, 375)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 45/45 [00:00<00:00, 80.61it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2350673, 375)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AF</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>TYPE</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>ADP</th>\n",
       "      <th>WT</th>\n",
       "      <th>HET</th>\n",
       "      <th>HOM</th>\n",
       "      <th>NC</th>\n",
       "      <th>DF_p1.GT</th>\n",
       "      <th>DF_p1.GQ</th>\n",
       "      <th>DF_p1.SDP</th>\n",
       "      <th>DF_p1.DP</th>\n",
       "      <th>DF_p1.FREQ</th>\n",
       "      <th>DF_p1.PVAL</th>\n",
       "      <th>DF_p1.AD</th>\n",
       "      <th>DF_p1.RD</th>\n",
       "      <th>DF_p2.GT</th>\n",
       "      <th>DF_p2.GQ</th>\n",
       "      <th>DF_p2.SDP</th>\n",
       "      <th>DF_p2.DP</th>\n",
       "      <th>DF_p2.FREQ</th>\n",
       "      <th>DF_p2.PVAL</th>\n",
       "      <th>DF_p2.AD</th>\n",
       "      <th>DF_p2.RD</th>\n",
       "      <th>DF_p23.GT</th>\n",
       "      <th>DF_p23.GQ</th>\n",
       "      <th>DF_p23.SDP</th>\n",
       "      <th>DF_p23.DP</th>\n",
       "      <th>DF_p23.FREQ</th>\n",
       "      <th>DF_p23.PVAL</th>\n",
       "      <th>DF_p23.AD</th>\n",
       "      <th>DF_p23.RD</th>\n",
       "      <th>DF_p24.GT</th>\n",
       "      <th>DF_p24.GQ</th>\n",
       "      <th>DF_p24.SDP</th>\n",
       "      <th>DF_p24.DP</th>\n",
       "      <th>DF_p24.FREQ</th>\n",
       "      <th>DF_p24.PVAL</th>\n",
       "      <th>DF_p24.AD</th>\n",
       "      <th>DF_p24.RD</th>\n",
       "      <th>DF_p25.GT</th>\n",
       "      <th>DF_p25.GQ</th>\n",
       "      <th>DF_p25.SDP</th>\n",
       "      <th>DF_p25.DP</th>\n",
       "      <th>DF_p25.FREQ</th>\n",
       "      <th>...</th>\n",
       "      <th>DF_p93.GT</th>\n",
       "      <th>DF_p93.GQ</th>\n",
       "      <th>DF_p93.SDP</th>\n",
       "      <th>DF_p93.DP</th>\n",
       "      <th>DF_p93.FREQ</th>\n",
       "      <th>DF_p93.PVAL</th>\n",
       "      <th>DF_p93.AD</th>\n",
       "      <th>DF_p93.RD</th>\n",
       "      <th>DF_p95.GT</th>\n",
       "      <th>DF_p95.GQ</th>\n",
       "      <th>DF_p95.SDP</th>\n",
       "      <th>DF_p95.DP</th>\n",
       "      <th>DF_p95.FREQ</th>\n",
       "      <th>DF_p95.PVAL</th>\n",
       "      <th>DF_p95.AD</th>\n",
       "      <th>DF_p95.RD</th>\n",
       "      <th>DF_p96.GT</th>\n",
       "      <th>DF_p96.GQ</th>\n",
       "      <th>DF_p96.SDP</th>\n",
       "      <th>DF_p96.DP</th>\n",
       "      <th>DF_p96.FREQ</th>\n",
       "      <th>DF_p96.PVAL</th>\n",
       "      <th>DF_p96.AD</th>\n",
       "      <th>DF_p96.RD</th>\n",
       "      <th>DF_p97.GT</th>\n",
       "      <th>DF_p97.GQ</th>\n",
       "      <th>DF_p97.SDP</th>\n",
       "      <th>DF_p97.DP</th>\n",
       "      <th>DF_p97.FREQ</th>\n",
       "      <th>DF_p97.PVAL</th>\n",
       "      <th>DF_p97.AD</th>\n",
       "      <th>DF_p97.RD</th>\n",
       "      <th>DF_p98.GT</th>\n",
       "      <th>DF_p98.GQ</th>\n",
       "      <th>DF_p98.SDP</th>\n",
       "      <th>DF_p98.DP</th>\n",
       "      <th>DF_p98.FREQ</th>\n",
       "      <th>DF_p98.PVAL</th>\n",
       "      <th>DF_p98.AD</th>\n",
       "      <th>DF_p98.RD</th>\n",
       "      <th>DF_p17.GT</th>\n",
       "      <th>DF_p17.GQ</th>\n",
       "      <th>DF_p17.SDP</th>\n",
       "      <th>DF_p17.DP</th>\n",
       "      <th>DF_p17.FREQ</th>\n",
       "      <th>DF_p17.PVAL</th>\n",
       "      <th>DF_p17.AD</th>\n",
       "      <th>DF_p17.RD</th>\n",
       "      <th>locus</th>\n",
       "      <th>MAF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>jcf7190000000000</td>\n",
       "      <td>77738</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.668265</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>SNP</td>\n",
       "      <td>PASS</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>49</td>\n",
       "      <td>28</td>\n",
       "      <td>8</td>\n",
       "      <td>T/C</td>\n",
       "      <td>49.0</td>\n",
       "      <td>17</td>\n",
       "      <td>17.0</td>\n",
       "      <td>70.59%</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>12.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T/C</td>\n",
       "      <td>18.0</td>\n",
       "      <td>8</td>\n",
       "      <td>8.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.012821</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>32.0</td>\n",
       "      <td>11</td>\n",
       "      <td>11.0</td>\n",
       "      <td>72.73%</td>\n",
       "      <td>5.160000e-04</td>\n",
       "      <td>8.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>29.0</td>\n",
       "      <td>15</td>\n",
       "      <td>15.0</td>\n",
       "      <td>53.33%</td>\n",
       "      <td>...</td>\n",
       "      <td>T/C</td>\n",
       "      <td>46.0</td>\n",
       "      <td>20</td>\n",
       "      <td>20.0</td>\n",
       "      <td>60%</td>\n",
       "      <td>2.254800e-05</td>\n",
       "      <td>12.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>40.0</td>\n",
       "      <td>15</td>\n",
       "      <td>15.0</td>\n",
       "      <td>66.67%</td>\n",
       "      <td>9.995000e-05</td>\n",
       "      <td>10.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>50.0</td>\n",
       "      <td>22</td>\n",
       "      <td>22.0</td>\n",
       "      <td>59.09%</td>\n",
       "      <td>9.581300e-06</td>\n",
       "      <td>13.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>60.0</td>\n",
       "      <td>18</td>\n",
       "      <td>18.0</td>\n",
       "      <td>77.78%</td>\n",
       "      <td>8.060500e-07</td>\n",
       "      <td>14.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T/C</td>\n",
       "      <td>32.0</td>\n",
       "      <td>11</td>\n",
       "      <td>11.0</td>\n",
       "      <td>72.73%</td>\n",
       "      <td>0.000516</td>\n",
       "      <td>8.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>jcf7190000000000-77738</td>\n",
       "      <td>0.331735</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>jcf7190000000000</td>\n",
       "      <td>77764</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.753734</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>SNP</td>\n",
       "      <td>PASS</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>28</td>\n",
       "      <td>50</td>\n",
       "      <td>8</td>\n",
       "      <td>T/C</td>\n",
       "      <td>50.0</td>\n",
       "      <td>16</td>\n",
       "      <td>16.0</td>\n",
       "      <td>75%</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>12.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C/C</td>\n",
       "      <td>49.0</td>\n",
       "      <td>13</td>\n",
       "      <td>13.0</td>\n",
       "      <td>84.62%</td>\n",
       "      <td>1.009600e-05</td>\n",
       "      <td>11.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>34.0</td>\n",
       "      <td>15</td>\n",
       "      <td>15.0</td>\n",
       "      <td>60%</td>\n",
       "      <td>...</td>\n",
       "      <td>T/C</td>\n",
       "      <td>53.0</td>\n",
       "      <td>19</td>\n",
       "      <td>19.0</td>\n",
       "      <td>68.42%</td>\n",
       "      <td>5.010600e-06</td>\n",
       "      <td>13.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>46.0</td>\n",
       "      <td>15</td>\n",
       "      <td>15.0</td>\n",
       "      <td>73.33%</td>\n",
       "      <td>2.498800e-05</td>\n",
       "      <td>11.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>75.0</td>\n",
       "      <td>25</td>\n",
       "      <td>25.0</td>\n",
       "      <td>72%</td>\n",
       "      <td>2.662600e-08</td>\n",
       "      <td>18.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>C/C</td>\n",
       "      <td>65.0</td>\n",
       "      <td>16</td>\n",
       "      <td>16.0</td>\n",
       "      <td>87.5%</td>\n",
       "      <td>2.545400e-07</td>\n",
       "      <td>14.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C/C</td>\n",
       "      <td>44.0</td>\n",
       "      <td>12</td>\n",
       "      <td>12.0</td>\n",
       "      <td>83.33%</td>\n",
       "      <td>0.000034</td>\n",
       "      <td>10.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>jcf7190000000000-77764</td>\n",
       "      <td>0.246266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>jcf7190000000004</td>\n",
       "      <td>54418</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.367180</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>SNP</td>\n",
       "      <td>PASS</td>\n",
       "      <td>36</td>\n",
       "      <td>1</td>\n",
       "      <td>86</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>39.0</td>\n",
       "      <td>47</td>\n",
       "      <td>47.0</td>\n",
       "      <td>25.53%</td>\n",
       "      <td>0.000109</td>\n",
       "      <td>12.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>27.0</td>\n",
       "      <td>20</td>\n",
       "      <td>20.0</td>\n",
       "      <td>40%</td>\n",
       "      <td>0.001638</td>\n",
       "      <td>8.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>26.0</td>\n",
       "      <td>31</td>\n",
       "      <td>31.0</td>\n",
       "      <td>25.81%</td>\n",
       "      <td>0.002333</td>\n",
       "      <td>8.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>32.0</td>\n",
       "      <td>43</td>\n",
       "      <td>43.0</td>\n",
       "      <td>23.26%</td>\n",
       "      <td>5.414800e-04</td>\n",
       "      <td>10.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>47.0</td>\n",
       "      <td>42</td>\n",
       "      <td>42.0</td>\n",
       "      <td>33.33%</td>\n",
       "      <td>...</td>\n",
       "      <td>T/C</td>\n",
       "      <td>72.0</td>\n",
       "      <td>45</td>\n",
       "      <td>45.0</td>\n",
       "      <td>44.44%</td>\n",
       "      <td>6.217800e-08</td>\n",
       "      <td>20.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>83.0</td>\n",
       "      <td>50</td>\n",
       "      <td>50.0</td>\n",
       "      <td>46%</td>\n",
       "      <td>4.345100e-09</td>\n",
       "      <td>23.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>29.0</td>\n",
       "      <td>36</td>\n",
       "      <td>36.0</td>\n",
       "      <td>25%</td>\n",
       "      <td>1.106100e-03</td>\n",
       "      <td>9.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>68.0</td>\n",
       "      <td>43</td>\n",
       "      <td>43.0</td>\n",
       "      <td>45.24%</td>\n",
       "      <td>1.352500e-07</td>\n",
       "      <td>19.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>T/C</td>\n",
       "      <td>76.0</td>\n",
       "      <td>45</td>\n",
       "      <td>45.0</td>\n",
       "      <td>46.67%</td>\n",
       "      <td>2.220600e-08</td>\n",
       "      <td>21.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>T/T</td>\n",
       "      <td>9.0</td>\n",
       "      <td>14</td>\n",
       "      <td>14.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.111110</td>\n",
       "      <td>3.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>jcf7190000000004-54418</td>\n",
       "      <td>0.367180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>jcf7190000000004</td>\n",
       "      <td>54508</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.080056</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>SNP</td>\n",
       "      <td>PASS</td>\n",
       "      <td>53</td>\n",
       "      <td>67</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>A/A</td>\n",
       "      <td>89.0</td>\n",
       "      <td>65</td>\n",
       "      <td>65.0</td>\n",
       "      <td>4.62%</td>\n",
       "      <td>0.122090</td>\n",
       "      <td>3.0</td>\n",
       "      <td>62.0</td>\n",
       "      <td>A/A</td>\n",
       "      <td>40.0</td>\n",
       "      <td>32</td>\n",
       "      <td>32.0</td>\n",
       "      <td>6.25%</td>\n",
       "      <td>0.246030</td>\n",
       "      <td>2.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>22.0</td>\n",
       "      <td>44</td>\n",
       "      <td>44.0</td>\n",
       "      <td>15.91%</td>\n",
       "      <td>0.006036</td>\n",
       "      <td>7.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>21.0</td>\n",
       "      <td>65</td>\n",
       "      <td>65.0</td>\n",
       "      <td>10.77%</td>\n",
       "      <td>6.590400e-03</td>\n",
       "      <td>7.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>15.0</td>\n",
       "      <td>67</td>\n",
       "      <td>67.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>A/A</td>\n",
       "      <td>85.0</td>\n",
       "      <td>67</td>\n",
       "      <td>67.0</td>\n",
       "      <td>5.97%</td>\n",
       "      <td>5.969100e-02</td>\n",
       "      <td>4.0</td>\n",
       "      <td>63.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>15.0</td>\n",
       "      <td>82</td>\n",
       "      <td>82.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.935100e-02</td>\n",
       "      <td>5.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>A/A</td>\n",
       "      <td>126.0</td>\n",
       "      <td>68</td>\n",
       "      <td>68.0</td>\n",
       "      <td>0%</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>22.0</td>\n",
       "      <td>52</td>\n",
       "      <td>52.0</td>\n",
       "      <td>13.46%</td>\n",
       "      <td>6.297700e-03</td>\n",
       "      <td>7.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>35.0</td>\n",
       "      <td>59</td>\n",
       "      <td>59.0</td>\n",
       "      <td>18.64%</td>\n",
       "      <td>2.925000e-04</td>\n",
       "      <td>11.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>A/A</td>\n",
       "      <td>28.0</td>\n",
       "      <td>28</td>\n",
       "      <td>28.0</td>\n",
       "      <td>10.71%</td>\n",
       "      <td>0.118180</td>\n",
       "      <td>3.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>jcf7190000000004-54508</td>\n",
       "      <td>0.080056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>jcf7190000000004</td>\n",
       "      <td>54540</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.333741</td>\n",
       "      <td>-10.0</td>\n",
       "      <td>SNP</td>\n",
       "      <td>PASS</td>\n",
       "      <td>56</td>\n",
       "      <td>1</td>\n",
       "      <td>58</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>56.0</td>\n",
       "      <td>67</td>\n",
       "      <td>67.0</td>\n",
       "      <td>25.37%</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>17.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>37.0</td>\n",
       "      <td>36</td>\n",
       "      <td>36.0</td>\n",
       "      <td>30.56%</td>\n",
       "      <td>0.000199</td>\n",
       "      <td>11.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>18.0</td>\n",
       "      <td>47</td>\n",
       "      <td>47.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.013188</td>\n",
       "      <td>6.0</td>\n",
       "      <td>41.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>83.0</td>\n",
       "      <td>67</td>\n",
       "      <td>67.0</td>\n",
       "      <td>35.82%</td>\n",
       "      <td>4.810900e-09</td>\n",
       "      <td>24.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>86.0</td>\n",
       "      <td>70</td>\n",
       "      <td>70.0</td>\n",
       "      <td>35.71%</td>\n",
       "      <td>...</td>\n",
       "      <td>A/G</td>\n",
       "      <td>67.0</td>\n",
       "      <td>67</td>\n",
       "      <td>67.0</td>\n",
       "      <td>29.85%</td>\n",
       "      <td>1.799800e-07</td>\n",
       "      <td>20.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>58.0</td>\n",
       "      <td>82</td>\n",
       "      <td>82.0</td>\n",
       "      <td>21.95%</td>\n",
       "      <td>1.339100e-06</td>\n",
       "      <td>18.0</td>\n",
       "      <td>64.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>45.0</td>\n",
       "      <td>69</td>\n",
       "      <td>69.0</td>\n",
       "      <td>20.29%</td>\n",
       "      <td>2.934400e-05</td>\n",
       "      <td>14.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>53.0</td>\n",
       "      <td>58</td>\n",
       "      <td>58.0</td>\n",
       "      <td>27.59%</td>\n",
       "      <td>4.601500e-06</td>\n",
       "      <td>16.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>88.0</td>\n",
       "      <td>62</td>\n",
       "      <td>62.0</td>\n",
       "      <td>40.32%</td>\n",
       "      <td>1.416800e-09</td>\n",
       "      <td>25.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>A/G</td>\n",
       "      <td>15.0</td>\n",
       "      <td>26</td>\n",
       "      <td>26.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.025310</td>\n",
       "      <td>5.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>jcf7190000000004-54540</td>\n",
       "      <td>0.333741</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 375 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              CHROM    POS REF ALT        AF  QUAL TYPE FILTER  ADP  WT  HET  \\\n",
       "0  jcf7190000000000  77738   T   C  0.668265 -10.0  SNP   PASS   13   2   49   \n",
       "1  jcf7190000000000  77764   T   C  0.753734 -10.0  SNP   PASS   12   1   28   \n",
       "3  jcf7190000000004  54418   T   C  0.367180 -10.0  SNP   PASS   36   1   86   \n",
       "5  jcf7190000000004  54508   A   G  0.080056 -10.0  SNP   PASS   53  67   20   \n",
       "6  jcf7190000000004  54540   A   G  0.333741 -10.0  SNP   PASS   56   1   58   \n",
       "\n",
       "   HOM  NC DF_p1.GT  DF_p1.GQ  DF_p1.SDP  DF_p1.DP DF_p1.FREQ  DF_p1.PVAL  \\\n",
       "0   28   8      T/C      49.0         17      17.0     70.59%    0.000011   \n",
       "1   50   8      T/C      50.0         16      16.0        75%    0.000008   \n",
       "3    0   0      T/C      39.0         47      47.0     25.53%    0.000109   \n",
       "5    0   0      A/A      89.0         65      65.0      4.62%    0.122090   \n",
       "6   28   0      A/G      56.0         67      67.0     25.37%    0.000002   \n",
       "\n",
       "   DF_p1.AD  DF_p1.RD DF_p2.GT  DF_p2.GQ  DF_p2.SDP  DF_p2.DP DF_p2.FREQ  \\\n",
       "0      12.0       5.0      ./.       NaN          6       NaN        NaN   \n",
       "1      12.0       4.0      ./.       NaN          4       NaN        NaN   \n",
       "3      12.0      35.0      T/C      27.0         20      20.0        40%   \n",
       "5       3.0      62.0      A/A      40.0         32      32.0      6.25%   \n",
       "6      17.0      50.0      A/G      37.0         36      36.0     30.56%   \n",
       "\n",
       "   DF_p2.PVAL  DF_p2.AD  DF_p2.RD DF_p23.GT  DF_p23.GQ  DF_p23.SDP  DF_p23.DP  \\\n",
       "0         NaN       NaN       NaN       T/C       18.0           8        8.0   \n",
       "1         NaN       NaN       NaN       ./.        NaN           6        NaN   \n",
       "3    0.001638       8.0      12.0       T/C       26.0          31       31.0   \n",
       "5    0.246030       2.0      30.0       A/G       22.0          44       44.0   \n",
       "6    0.000199      11.0      25.0       A/G       18.0          47       47.0   \n",
       "\n",
       "  DF_p23.FREQ  DF_p23.PVAL  DF_p23.AD  DF_p23.RD DF_p24.GT  DF_p24.GQ  \\\n",
       "0         NaN     0.012821        5.0        3.0       T/C       32.0   \n",
       "1         NaN          NaN        NaN        NaN       C/C       49.0   \n",
       "3      25.81%     0.002333        8.0       23.0       T/C       32.0   \n",
       "5      15.91%     0.006036        7.0       37.0       A/G       21.0   \n",
       "6         NaN     0.013188        6.0       41.0       A/G       83.0   \n",
       "\n",
       "   DF_p24.SDP  DF_p24.DP DF_p24.FREQ   DF_p24.PVAL  DF_p24.AD  DF_p24.RD  \\\n",
       "0          11       11.0      72.73%  5.160000e-04        8.0        3.0   \n",
       "1          13       13.0      84.62%  1.009600e-05       11.0        2.0   \n",
       "3          43       43.0      23.26%  5.414800e-04       10.0       33.0   \n",
       "5          65       65.0      10.77%  6.590400e-03        7.0       58.0   \n",
       "6          67       67.0      35.82%  4.810900e-09       24.0       43.0   \n",
       "\n",
       "  DF_p25.GT  DF_p25.GQ  DF_p25.SDP  DF_p25.DP DF_p25.FREQ  ...  DF_p93.GT  \\\n",
       "0       T/C       29.0          15       15.0      53.33%  ...        T/C   \n",
       "1       T/C       34.0          15       15.0         60%  ...        T/C   \n",
       "3       T/C       47.0          42       42.0      33.33%  ...        T/C   \n",
       "5       A/G       15.0          67       67.0         NaN  ...        A/A   \n",
       "6       A/G       86.0          70       70.0      35.71%  ...        A/G   \n",
       "\n",
       "   DF_p93.GQ  DF_p93.SDP DF_p93.DP  DF_p93.FREQ   DF_p93.PVAL  DF_p93.AD  \\\n",
       "0       46.0          20      20.0          60%  2.254800e-05       12.0   \n",
       "1       53.0          19      19.0       68.42%  5.010600e-06       13.0   \n",
       "3       72.0          45      45.0       44.44%  6.217800e-08       20.0   \n",
       "5       85.0          67      67.0        5.97%  5.969100e-02        4.0   \n",
       "6       67.0          67      67.0       29.85%  1.799800e-07       20.0   \n",
       "\n",
       "  DF_p93.RD  DF_p95.GT  DF_p95.GQ  DF_p95.SDP DF_p95.DP  DF_p95.FREQ  \\\n",
       "0       8.0        T/C       40.0          15      15.0       66.67%   \n",
       "1       6.0        T/C       46.0          15      15.0       73.33%   \n",
       "3      25.0        T/C       83.0          50      50.0          46%   \n",
       "5      63.0        A/G       15.0          82      82.0          NaN   \n",
       "6      47.0        A/G       58.0          82      82.0       21.95%   \n",
       "\n",
       "    DF_p95.PVAL  DF_p95.AD DF_p95.RD  DF_p96.GT  DF_p96.GQ  DF_p96.SDP  \\\n",
       "0  9.995000e-05       10.0       5.0        T/C       50.0          22   \n",
       "1  2.498800e-05       11.0       4.0        T/C       75.0          25   \n",
       "3  4.345100e-09       23.0      27.0        T/C       29.0          36   \n",
       "5  2.935100e-02        5.0      77.0        A/A      126.0          68   \n",
       "6  1.339100e-06       18.0      64.0        A/G       45.0          69   \n",
       "\n",
       "  DF_p96.DP  DF_p96.FREQ   DF_p96.PVAL  DF_p96.AD DF_p96.RD  DF_p97.GT  \\\n",
       "0      22.0       59.09%  9.581300e-06       13.0       9.0        T/C   \n",
       "1      25.0          72%  2.662600e-08       18.0       7.0        C/C   \n",
       "3      36.0          25%  1.106100e-03        9.0      27.0        T/C   \n",
       "5      68.0           0%  1.000000e+00        0.0      68.0        A/G   \n",
       "6      69.0       20.29%  2.934400e-05       14.0      55.0        A/G   \n",
       "\n",
       "   DF_p97.GQ  DF_p97.SDP DF_p97.DP  DF_p97.FREQ   DF_p97.PVAL  DF_p97.AD  \\\n",
       "0       60.0          18      18.0       77.78%  8.060500e-07       14.0   \n",
       "1       65.0          16      16.0        87.5%  2.545400e-07       14.0   \n",
       "3       68.0          43      43.0       45.24%  1.352500e-07       19.0   \n",
       "5       22.0          52      52.0       13.46%  6.297700e-03        7.0   \n",
       "6       53.0          58      58.0       27.59%  4.601500e-06       16.0   \n",
       "\n",
       "  DF_p97.RD  DF_p98.GT  DF_p98.GQ  DF_p98.SDP DF_p98.DP  DF_p98.FREQ  \\\n",
       "0       4.0        ./.        NaN           6       NaN          NaN   \n",
       "1       2.0        ./.        NaN           4       NaN          NaN   \n",
       "3      24.0        T/C       76.0          45      45.0       46.67%   \n",
       "5      45.0        A/G       35.0          59      59.0       18.64%   \n",
       "6      42.0        A/G       88.0          62      62.0       40.32%   \n",
       "\n",
       "    DF_p98.PVAL  DF_p98.AD DF_p98.RD  DF_p17.GT  DF_p17.GQ  DF_p17.SDP  \\\n",
       "0           NaN        NaN       NaN        T/C       32.0          11   \n",
       "1           NaN        NaN       NaN        C/C       44.0          12   \n",
       "3  2.220600e-08       21.0      24.0        T/T        9.0          14   \n",
       "5  2.925000e-04       11.0      48.0        A/A       28.0          28   \n",
       "6  1.416800e-09       25.0      37.0        A/G       15.0          26   \n",
       "\n",
       "  DF_p17.DP  DF_p17.FREQ  DF_p17.PVAL  DF_p17.AD DF_p17.RD  \\\n",
       "0      11.0       72.73%     0.000516        8.0       3.0   \n",
       "1      12.0       83.33%     0.000034       10.0       2.0   \n",
       "3      14.0          NaN     0.111110        3.0      11.0   \n",
       "5      28.0       10.71%     0.118180        3.0      25.0   \n",
       "6      26.0          NaN     0.025310        5.0      21.0   \n",
       "\n",
       "                    locus       MAF  \n",
       "0  jcf7190000000000-77738  0.331735  \n",
       "1  jcf7190000000000-77764  0.246266  \n",
       "3  jcf7190000000004-54418  0.367180  \n",
       "5  jcf7190000000004-54508  0.080056  \n",
       "6  jcf7190000000004-54540  0.333741  \n",
       "\n",
       "[5 rows x 375 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(coastal.shape)\n",
    "coastal_recalc = recalc_rd(coastal)\n",
    "print(coastal_recalc.shape)\n",
    "coastal_recalc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "# save coastal\n",
    "coastfile = op.join(newdirs['FDC'], 'DF_pooled-varscan_all_bedfiles_SNP_FDC_maf_RD-recalculated.txt')\n",
    "write_jobs.append(lview.apply_async(save_file, *(coastal_recalc, coastfile)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 45/45 [00:00<00:00, 3780.47it/s]\n",
      "100%|██████████| 45/45 [00:00<00:00, 1586.17it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DF_pooled-varscan_all_bedfiles_PARALOGS_FDC.txt (1110, 375)\n",
      "\t (1110, 375)\n",
      "DF_pooled-varscan_all_bedfiles_REPEATS_FDC.txt (79955, 375)\n",
      "\t (79955, 375)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "# recalc RD for paralogs and repeats\n",
    "fdc_recalc = {}\n",
    "for key,df in fdc_remaining.items():\n",
    "    print(op.basename(key), df.shape)\n",
    "    fdc_recalc[key] = recalc_rd(df)\n",
    "    print('\\t', fdc_recalc[key].shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DF_pooled-varscan_all_bedfiles_PARALOGS_FDC_maf_RD-recalculated.txt\n",
      "DF_pooled-varscan_all_bedfiles_REPEATS_FDC_maf_RD-recalculated.txt\n"
     ]
    }
   ],
   "source": [
    "# save paralog and repeats\n",
    "for key,df in fdc_recalc.items():\n",
    "    dst = op.join(newdirs['FDC'], op.basename(key).replace(\".txt\", \"_maf_RD-recalculated.txt\"))\n",
    "    print(op.basename(dst))\n",
    "    save_file(df, dst)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/data/projects/pool_seq/DF_datasets/DF_pooled_GEA/DF_pooled/snpsANDindels/03_maf-p05_RD-recalculated_FDC/DF_pooled-varscan_all_bedfiles_REPEATS_FDC_maf_RD-recalculated.txt'"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dst"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
